---
vignette: >
  % \VignetteIndexEntry{Transcriptome-Proteome analysis}
  % \VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
output:
  html_document:
    mathjax: null   
---

# Setup

```{r, message = FALSE}
library(BioPlex)
library(AnnotationDbi)
library(AnnotationHub)
library(ExperimentHub)
library(ExpressionAtlas)
library(GenomicFeatures)
library(edgeR)
library(ggplot2)
library(limma)
```

# CCLE transcriptome and proteome data for HCT116

Get CCLE transcriptome data for HCT116:
 
```{r ccle-transcriptome}
atlasRes <- ExpressionAtlas::searchAtlasExperiments(
              properties = "Cancer Cell Line Encyclopedia", 
              species = "human" )
atlasRes
ccle.trans <- ExpressionAtlas::getAtlasExperiment("E-MTAB-2770")
ccle.trans <- ccle.trans[[1]]
ccle.trans <- ccle.trans[,grep("HCT 116", ccle.trans$cell_line)]
ccle.trans
```

There is currently an
[issue with obtaining E-MTAB-2770](https://github.com/ebi-gene-expression-group/bioconductor-ExpressionAtlas/issues/6) via the
[ExpressionAtlas](https://bioconductor.org/packages/ExpressionAtlas) package.
We therefore pull the file directly from ftp as a workaround.

```{r, eval = FALSE}
ebi.ftp.url <- "ftp.ebi.ac.uk/pub/databases/microarray/data/atlas/experiments"
mtab.url <- "E-MTAB-2770/archive/E-MTAB-2770-atlasExperimentSummary.Rdata.1"
mtab.url <- file.path(ebi.ftp.url, mtab.url)
download.file(mtab.url, "E-MTAB-2770.Rdata")
load("E-MTAB-2770.Rdata")
ccle.trans <- experimentSummary
file.remove("E-MTAB-2770.Rdata")
```

```{r E-MTAB-2770-cache, echo = FALSE}
ccle.trans <- BioPlexAnalysis:::.getResourceFromCache("ccle.trans", update.value = NA)
if(is.null(ccle.trans))
{
    ebi.ftp.url <- "ftp.ebi.ac.uk/pub/databases/microarray/data/atlas/experiments"
    mtab.url <- "E-MTAB-2770/archive/E-MTAB-2770-atlasExperimentSummary.Rdata.1"
    mtab.url <- file.path(ebi.ftp.url, mtab.url)
    download.file(mtab.url, "E-MTAB-2770.Rdata")
    load("E-MTAB-2770.Rdata")
    ccle.trans <- experimentSummary
    file.remove("E-MTAB-2770.Rdata")
    BioPlexAnalysis:::.cacheResource(ccle.trans, "ccle.trans")
}
```

We proceed as before:

```{r}
ccle.trans <- ccle.trans[[1]]
ccle.trans <- ccle.trans[,grep("HCT 116", ccle.trans$cell_line)]
ccle.trans
```

Get the CCLE proteome data for HCT116:

```{r ccle-proteome}
eh <- ExperimentHub::ExperimentHub()
AnnotationHub::query(eh, c("gygi", "depmap"))
ccle.prot <- eh[["EH3459"]]
ccle.prot <- as.data.frame(ccle.prot)
ccle.prot <- BioPlex::ccleProteome2SummarizedExperiment(ccle.prot)
ccle.prot
```

Connect to 
[AnnotationHub](http://bioconductor.org/packages/AnnotationHub) and
obtain OrgDb package for human:

```{r ahub, message = FALSE}
ah <- AnnotationHub::AnnotationHub()
orgdb <- AnnotationHub::query(ah, c("orgDb", "Homo sapiens"))
orgdb <- orgdb[[1]]
```

Map to ENSEMBL for comparison with CCLE transcriptome data for HCT116:

```{r}
rnames <- AnnotationDbi::mapIds(orgdb,
                      keytype = "UNIPROT",
                      column = "ENSEMBL",
                      keys = rownames(ccle.prot))
```

Subset to the ENSEMBL IDs that both datasets have in common

```{r}
isect <- intersect(rnames, rownames(ccle.trans))
ind <- match(isect, rnames)
```

This should be rather RPKM, provided gene length from EDASeq:
 
```{r}
assay(ccle.trans, "cpm") <- edgeR::cpm(assay(ccle.trans), log = TRUE)
```

A look at general correlation between transcriptome and proteome:

```{r}
cor.test(assay(ccle.trans, "cpm")[isect,], 
         assay(ccle.prot)[ind,],
         use = "complete.obs")
```

```{r}
df <- data.frame(trans = assay(ccle.trans, "cpm")[isect,],
                 prot = assay(ccle.prot)[ind,])
ggplot(df, aes(x = trans, y = prot) ) +
    geom_bin2d(bins = 70) +
    scale_fill_continuous(type = "viridis") +
    xlab("log2 CPM") +
    ylab("log2 intensity") +
    theme_bw()
```


Let's check whether this looks very different when accounting for gene length.
We therefore obtain gene length for the hg38 genome assembly (used for CCLE).

```{r, message = FALSE}
AnnotationHub::query(ah, c("TxDb", "Homo sapiens"))
txdb <- ah[["AH92591"]]
gs <- GenomicFeatures::genes(txdb)
gs
len <- GenomicRanges::width(gs)
names(len) <- names(gs)
head(len)
```

This requires to map from Entrez IDs present for the gene length data to ENSEMBL IDs
present in the transcriptomic data. 

```{r}
eids <- AnnotationDbi::mapIds(orgdb,
                              column = "ENTREZID",
                              keytype = "ENSEMBL", 
                              keys = rownames(ccle.trans))
rowData(ccle.trans)$length <- len[eids]
```

We can now compute RPKM given the obtained gene lengths as input.

```{r}
assay(ccle.trans, "rpkm") <- edgeR::rpkm(assay(ccle.trans), 
                                         gene.length = rowData(ccle.trans)$length,
                                         log = TRUE) 
```

```{r}
cor.test(assay(ccle.trans, "rpkm")[isect,],
         assay(ccle.prot)[ind,],
         use = "complete.obs")
```

```{r}
df <- data.frame(trans = assay(ccle.trans, "rpkm")[isect,],
                 prot = assay(ccle.prot)[ind,])
ggplot(df, aes(x = trans, y = prot) ) +
    geom_bin2d(bins = 70) +
    scale_fill_continuous(type = "viridis") +
    xlab("log2 RPKM") +
    ylab("log2 intensity") +
    theme_bw()
```

# DE analysis HEK293 vs. HCT116 (transcriptomic and proteomic level)

Pull the HEK293 data:

```{r}
gse.293t <- BioPlex::getGSE122425()
```

Pull the HCT116 data:

```{r}
klijn <- ExpressionAtlas::getAtlasData("E-MTAB-2706")
klijn <- klijn$`E-MTAB-2706`$rnaseq
klijn
```

Combine the both HCT116 samples:

```{r}
ind2 <- grep("HCT 116", klijn$cell_line)
isect <- intersect(rownames(ccle.trans), rownames(klijn))
emat <- cbind(assay(ccle.trans)[isect,], assay(klijn)[isect,ind2])
colnames(emat) <- c("ccle", "klijn")
head(emat)
```

Combine with the HEK293 wildtype samples:

```{r}
isect <- intersect(rownames(emat), rownames(gse.293t))
emat <- cbind(emat[isect,], assay(gse.293t)[isect, 1:3])
colnames(emat) <- paste0(rep(c("HCT", "HEK"), c(2,3)), c(1:2, 1:3)) 
```

Compute logCPMs to bring samples from different cell lines and experiments on
the same scale using the limma-trend approach:

```{r}
dge <- edgeR::DGEList(counts = emat)
dge$group <- rep(c("HCT", "HEK"), c(2,3))
design <- model.matrix(~ dge$group)
keep <- edgeR::filterByExpr(dge, design)
dge <- dge[keep,,keep.lib.sizes = FALSE]
dim(dge)
```

```{r}
dge <- edgeR::calcNormFactors(dge)
logCPM <- edgeR::cpm(dge, log = TRUE, prior.count = 3)
fit <- limma::lmFit(logCPM, design)
fit <- limma::eBayes(fit, trend = TRUE)
limma::topTable(fit, coef = ncol(design))
tt <- limma::topTable(fit, coef = ncol(design), number = nrow(logCPM), sort.by = "none")
```

Now let's pull the BioPlex3 proteome data:

```{r}
bp.prot <- BioPlex::getBioplexProteome()
rowData(bp.prot)
```

Compare differential expression results on transcriptomic and proteomic level
based on gene symbols as those are readily available:

```{r}
isect <- intersect(rowData(bp.prot)$SYMBOL, 
                   rowData(gse.293t)[rownames(logCPM), "SYMBOL"])
length(isect)
```

```{r}
ind.trans <- match(isect, rowData(gse.293t)[rownames(logCPM), "SYMBOL"])
ind.prot <- match(isect, rowData(bp.prot)$SYMBOL)
```

We need to switch here the sign of the fold change because the transcriptome
is HEK-vs-HCT, the proteome is HCT-vs-HEK:

```{r}
cor.test(-1 * tt[ind.trans, "logFC"], 
         rowData(bp.prot)[ind.prot, "log2ratio"])
```

```{r}
df <- data.frame(trans = -1 * tt[ind.trans, "logFC"],
                 prot = rowData(bp.prot)[ind.prot, "log2ratio"])
ggplot(df, aes(x = trans, y = prot) ) +
    geom_bin2d(bins = 70) +
    scale_fill_continuous(type = "viridis") +
    xlab("log2FC (transcriptome)") +
    ylab("log2FC (proteome)") +
    theme_bw()
```

# Interaction networks of cell-line specific proteins

We can now inspect the interactions of proteins that are strongly differentially
expressed between cell lines as in Supplementary Figure S3, Panels J-M, of the
[BioPlex 3.0 publication](https://doi.org/10.1016/j.cell.2021.04.011). 

Here, we inspect the interactions of CDH2, a protein that was observed in
~5-fold lower abundance in the HCT116 cell line when compared to the 293T cell line. 

```{r}
subset(rowData(bp.prot), SYMBOL == "CDH2")
```

We therefore obtain the latest versions of the both BioPlex PPI networks ...

```{r}
bp.293t <- BioPlex::getBioPlex(cell.line = "293T", version = "3.0")
bp.hct <- BioPlex::getBioPlex(cell.line = "HCT116", version = "1.0")
```

Get all interactions involving CDH2:

```{r}
cdh2.293t <- subset(bp.293t, SymbolA == "CDH2" | SymbolB == "CDH2")
cdh2.293t
cdh2.hct <- subset(bp.hct, SymbolA == "CDH2" | SymbolB == "CDH2")
cdh2.hct 
```

Expand by including interactions between interactors of CDH2:

```{r}
cdh2i <- cdh2.293t$SymbolA
cdh2i.293t <- subset(bp.293t, SymbolA %in% cdh2i & SymbolB %in% cdh2i)
cdh2i.293t
cdh2i.hct <- subset(bp.hct, SymbolA %in% cdh2i & SymbolB %in% cdh2i)
cdh2i.hct
```

Now we construct a joined graph of interactions involving CDH2 or one of its
interactors for both networks:

```{r}
cdh2.293t <- rbind(cdh2.293t, cdh2i.293t)
cdh2.293t$CellLine <- "293T"
cdh2.hct <- cdh2i.hct
cdh2.hct$CellLine <- "HCT116"
cdh2.df <- rbind(cdh2.293t, cdh2.hct)
cdh2.df
```

We restrict here to the relevant columns for plotting:

```{r}
rel.cols <- c("SymbolA", "SymbolB", "CellLine")
cdh2.df <- cdh2.df[,rel.cols]
cdh2.df <- unique(cdh2.df)
cdh2.df
```
