<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title> • BioPlexAnalysis</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="">
<meta property="og:description" content="BioPlexAnalysis">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioPlexAnalysis</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.99.3</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BioNet.html">Maximum scoring subnetwork analysis</a>
    </li>
    <li>
      <a href="../articles/CORUM.html">Protein complex overlap analysis</a>
    </li>
    <li>
      <a href="../articles/PDB.html">Mapping PPIs onto 3D protein structures</a>
    </li>
    <li>
      <a href="../articles/PFAM.html">Protein domain association analysis</a>
    </li>
    <li>
      <a href="../articles/TranscriptomeProteome.html">Transcriptome and Proteome analysis</a>
    </li>
    <li>
      <a href="../articles/GraphFrames.html">GraphFrames backend</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ccb-hms/BioPlexAnalysis" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip></h1>
            
      
      
      <div class="hidden name"><code>BioNet.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="section-setup">Setup<a class="anchor" aria-label="anchor" href="#section-setup"></a>
</h2>
<div class="sourceCode" id="section-cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ccb-hms/BioPlex" class="external-link">BioPlex</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bionet.bioapps.biozentrum.uni-wuerzburg.de/" class="external-link">BioNet</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">EnrichmentBrowser</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">graph</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="section-bionet">BioNet<a class="anchor" aria-label="anchor" href="#section-bionet"></a>
</h2>
<p>Get the latest version of the 293T PPI network:</p>
<div class="sourceCode" id="section-cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.293t</span> <span class="op">&lt;-</span> <span class="fu">BioPlex</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioPlex/man/getBioPlex.html" class="external-link">getBioPlex</a></span><span class="op">(</span>cell.line <span class="op">=</span> <span class="st">"293T"</span>, version <span class="op">=</span> <span class="st">"3.0"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Using cached version from 2022-06-21 11:49:51</span></code></pre>
<p>and turn into a graph object:</p>
<div class="sourceCode" id="section-cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.gr</span> <span class="op">&lt;-</span> <span class="fu">BioPlex</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioPlex/man/bioplex2graph.html" class="external-link">bioplex2graph</a></span><span class="op">(</span><span class="va">bp.293t</span><span class="op">)</span></code></pre></div>
<p>Get RNA-seq data for 293T cells from GEO: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122425" class="external-link">GSE122425</a></p>
<div class="sourceCode" id="section-cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gse122425</span> <span class="op">&lt;-</span> <span class="fu">BioPlex</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioPlex/man/getGSE122425.html" class="external-link">getGSE122425</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Using cached version from 2022-06-21 11:50:01</span></code></pre>
<p>Differential expression analysis between NSUN2 knockout and wild
type:</p>
<div class="sourceCode" id="section-cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gse122425</span><span class="op">$</span><span class="va">GROUP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="va">gse122425</span> <span class="op">&lt;-</span> <span class="fu">EnrichmentBrowser</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/EnrichmentBrowser/man/deAna.html" class="external-link">deAna</a></span><span class="op">(</span><span class="va">gse122425</span>, de.method <span class="op">=</span> <span class="st">"DESeq2"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Excluding 38949 genes not satisfying min.cpm threshold</span></code></pre>
<pre><code><span class="co">## Corresponding software package not found: DESeq2</span>
<span class="co">## Make sure that you have it installed.</span></code></pre>
<pre><code><span class="co">## 'getOption("repos")' replaces Bioconductor standard repositories, see</span>
<span class="co">## '?repositories' for details</span>
<span class="co">## </span>
<span class="co">## replacement repositories:</span>
<span class="co">##     CRAN: https://packagemanager.rstudio.com/cran/__linux__/focal/latest</span></code></pre>
<pre><code><span class="co">## Bioconductor version 3.15 (BiocManager 1.30.18), R 4.2.0 (2022-04-22)</span></code></pre>
<pre><code><span class="co">## Installing package(s) 'DESeq2'</span></code></pre>
<pre><code><span class="co">## also installing the dependencies 'genefilter', 'geneplotter', 'RcppArmadillo'</span></code></pre>
<p>Inspect results:</p>
<div class="sourceCode" id="section-cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## DataFrame with 18956 rows and 8 columns</span>
<span class="co">##                        SYMBOL          KO                     GO    length</span>
<span class="co">##                   &lt;character&gt; &lt;character&gt;            &lt;character&gt; &lt;integer&gt;</span>
<span class="co">## ENSG00000227232        WASH7P      K18461                      _     15444</span>
<span class="co">## ENSG00000228463    AP006222.2           _                      _     39639</span>
<span class="co">## ENSG00000237094 RP4-669L17.10           _                      _    136229</span>
<span class="co">## ENSG00000225630      MTND2P28      K03879                      _      1044</span>
<span class="co">## ENSG00000237973  hsa-mir-6723      K02256                      _      1543</span>
<span class="co">## ...                       ...         ...                    ...       ...</span>
<span class="co">## ENSG00000185973         TMLHE      K00474 molecular_function:i..    179830</span>
<span class="co">## ENSG00000124333         VAMP7      K08515 biological_process:a..     62478</span>
<span class="co">## ENSG00000182484        WASH6P      K18461 molecular_function:m..      4885</span>
<span class="co">## ENSG00000215414       PSMA6P1      K02730                      _       741</span>
<span class="co">## ENSG00000185275        CD24P4      K06469                      _       243</span>
<span class="co">##                        FC DESeq2.STAT        PVAL    ADJ.PVAL</span>
<span class="co">##                 &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;</span>
<span class="co">## ENSG00000227232  0.171520    0.724534    0.468738    0.648277</span>
<span class="co">## ENSG00000228463  0.445551    1.194110    0.232435    0.419636</span>
<span class="co">## ENSG00000237094 -0.318485   -0.446073    0.655545    0.791683</span>
<span class="co">## ENSG00000225630 -0.404707   -1.227372    0.219683    0.404906</span>
<span class="co">## ENSG00000237973 -0.337735   -0.983072    0.325572    0.518622</span>
<span class="co">## ...                   ...         ...         ...         ...</span>
<span class="co">## ENSG00000185973  0.111457    0.861636 3.88888e-01 5.79293e-01</span>
<span class="co">## ENSG00000124333  3.430248    4.951910 7.34886e-07 2.15608e-05</span>
<span class="co">## ENSG00000182484 -0.247212   -0.398845 6.90007e-01 8.17000e-01</span>
<span class="co">## ENSG00000215414  0.543926    0.991335 3.21522e-01 5.14536e-01</span>
<span class="co">## ENSG00000185275  0.452405    1.596938 1.10280e-01 2.59482e-01</span></code></pre>
<div class="sourceCode" id="section-cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">not.na</span> <span class="op">&lt;-</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span><span class="op">$</span><span class="va">PVAL</span><span class="op">)</span>
<span class="va">gse122425</span> <span class="op">&lt;-</span> <span class="va">gse122425</span><span class="op">[</span><span class="va">not.na</span>,<span class="op">]</span>
<span class="fu">EnrichmentBrowser</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/EnrichmentBrowser/man/plots.html" class="external-link">volcano</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span><span class="op">$</span><span class="va">FC</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span><span class="op">$</span><span class="va">ADJ.PVAL</span><span class="op">)</span></code></pre></div>
<p><img src="BioNet_files/figure-html/de-results-gse122425-1.png" width="700"></p>
<p>Change rownames to gene symbols for intersection with bioplex
graph:</p>
<div class="sourceCode" id="section-cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span><span class="op">$</span><span class="va">ENSEMBL</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span><span class="op">$</span><span class="va">SYMBOL</span></code></pre></div>
<p>Intersect bioplex graph and expression dataset:</p>
<div class="sourceCode" id="section-cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">syms</span> <span class="op">&lt;-</span> <span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/nodeData-methods.html" class="external-link">nodeData</a></span><span class="op">(</span><span class="va">bp.gr</span>, <span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/graphNEL-class.html" class="external-link">nodes</a></span><span class="op">(</span><span class="va">bp.gr</span><span class="op">)</span>, <span class="st">"SYMBOL"</span><span class="op">)</span>
<span class="va">syms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="va">syms</span><span class="op">)</span>
<span class="va">isect</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">syms</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span><span class="op">)</span> 
<span class="va">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">isect</span>, <span class="va">syms</span><span class="op">)</span> 
<span class="va">bp.sgr</span> <span class="op">&lt;-</span> <span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/subGraph.html" class="external-link">subGraph</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">syms</span><span class="op">)</span><span class="op">[</span><span class="va">ind</span><span class="op">]</span>, <span class="va">bp.gr</span><span class="op">)</span>
<span class="va">bp.sgr</span> </code></pre></div>
<pre><code><span class="co">## A graphNEL graph with directed edges</span>
<span class="co">## Number of Nodes = 11015 </span>
<span class="co">## Number of Edges = 79341</span></code></pre>
<div class="sourceCode" id="section-cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.sgr</span> <span class="op">&lt;-</span> <span class="fu">BioNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioNet/man/rmSelfLoops.html" class="external-link">rmSelfLoops</a></span><span class="op">(</span><span class="va">bp.sgr</span><span class="op">)</span>
<span class="va">bp.sgr</span></code></pre></div>
<pre><code><span class="co">## A graphNEL graph with directed edges</span>
<span class="co">## Number of Nodes = 11015 </span>
<span class="co">## Number of Edges = 79341</span></code></pre>
<p>Get the DE p-values for each node:</p>
<div class="sourceCode" id="section-cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">syms</span> <span class="op">&lt;-</span> <span class="va">syms</span><span class="op">[</span><span class="va">ind</span><span class="op">]</span>
<span class="va">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">isect</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span><span class="op">)</span>
<span class="va">pvals</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span><span class="op">[</span><span class="va">ind</span>, <span class="st">"PVAL"</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">syms</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##     P00813     Q8N7W2     Q6ZMN8     P55039     Q9NZL4     Q6ICG6 </span>
<span class="co">## 0.57137317 0.05761896 0.04112662 0.22823806 0.84595685 0.38653260</span></code></pre>
<p>To score each node of the network we fit a Beta-uniform mixture model
(BUM) to the p-value distribution and subsequently use the parameters of
the model for the scoring function:</p>
<div class="sourceCode" id="section-cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fb</span> <span class="op">&lt;-</span> <span class="fu">BioNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioNet/man/fitBumModel.html" class="external-link">fitBumModel</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">)</span></code></pre></div>
<p><img src="BioNet_files/figure-html/bum-bp-gse122425-1.png" width="700"></p>
<div class="sourceCode" id="section-cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu">BioNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioNet/man/scoreNodes.html" class="external-link">scoreNodes</a></span><span class="op">(</span><span class="va">bp.sgr</span>, <span class="va">fb</span>, fdr <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">scores</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##     P00813     Q8N7W2     Q6ZMN8     P55039     Q9NZL4     Q6ICG6 </span>
<span class="co">## -2.3408848 -0.4505584 -0.1727217 -1.5847732 -2.6642288 -2.0188587</span></code></pre>
<div class="sourceCode" id="section-cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">scores</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##     Min.  1st Qu.   Median     Mean  3rd Qu.     Max. </span>
<span class="co">##  -2.8019  -2.1898  -1.2382   0.3586   0.6252 265.2463</span></code></pre>
<div class="section level3">
<h3 id="section-first-reduction-step-unsupervised-detection-of-maximum-scoring-subnetwork">First reduction step: unsupervised detection of maximum scoring
subnetwork<a class="anchor" aria-label="anchor" href="#section-first-reduction-step-unsupervised-detection-of-maximum-scoring-subnetwork"></a>
</h3>
<p>Here we use a fast heuristic approach to calculate an approximation
to theoptimal scoring subnetwork.<br>
An optimal solution can be calculated using the heinz algorithm
requiring a commercial CPLEX license.</p>
<p>Note: this will run for a couple of minutes so it will be a good idea
to cache the result to not recompute this over and over again.</p>
<div class="sourceCode" id="section-cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">module</span> <span class="op">&lt;-</span> <span class="fu">BioNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioNet/man/runFastHeinz.html" class="external-link">runFastHeinz</a></span><span class="op">(</span><span class="va">bp.sgr</span>, <span class="va">scores</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Using cached version from 2022-06-21 12:13:53</span></code></pre>
<div class="sourceCode" id="section-cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">module</span></code></pre></div>
<pre><code><span class="co">## A graphNEL graph with directed edges</span>
<span class="co">## Number of Nodes = 2851 </span>
<span class="co">## Number of Edges = 8524</span></code></pre>
<p>For some reason <code><a href="https://rdrr.io/pkg/BioNet/man/runFastHeinz.html" class="external-link">BioNet::runFastHeinz</a></code> drops the
<code>edgeData</code> of the input graph, so we need to re-annotate
those to the module. There doesn’t seem to be a convenient way to bulk
transfer <code>edgeData</code> from one graph to another, but rather
only for one edge and one attribute at a time.</p>
<p>With a little abuse, we help ourselves via accessing slots
directly.</p>
<div class="sourceCode" id="section-cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">en</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/edgeData-methods.html" class="external-link">edgeData</a></span><span class="op">(</span><span class="va">module</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/graph/man/edgeDataDefaults-methods.html" class="external-link">edgeDataDefaults</a></span><span class="op">(</span><span class="va">module</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/graph/man/edgeDataDefaults-methods.html" class="external-link">edgeDataDefaults</a></span><span class="op">(</span><span class="va">bp.sgr</span><span class="op">)</span>
<span class="va">module</span><span class="op">@</span><span class="va">edgeData</span><span class="op">@</span><span class="va">data</span><span class="op">[</span><span class="va">en</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/graph/man/edgeData-methods.html" class="external-link">edgeData</a></span><span class="op">(</span><span class="va">bp.sgr</span><span class="op">)</span><span class="op">[</span><span class="va">en</span><span class="op">]</span></code></pre></div>
</div>
<div class="section level3">
<h3 id="section-second-reduction-step-supervised-enrichment-analysis-of-maximum-scoring-subnetwork">Second reduction step: supervised enrichment analysis of maximum
scoring subnetwork<a class="anchor" aria-label="anchor" href="#section-second-reduction-step-supervised-enrichment-analysis-of-maximum-scoring-subnetwork"></a>
</h3>
<div class="sourceCode" id="section-cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">kegg.gs</span> <span class="op">&lt;-</span> <span class="fu">EnrichmentBrowser</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/EnrichmentBrowser/man/getGenesets.html" class="external-link">getGenesets</a></span><span class="op">(</span>org <span class="op">=</span> <span class="st">"hsa"</span>, db <span class="op">=</span> <span class="st">"kegg"</span>, gene.id.type <span class="op">=</span> <span class="st">"UNIPROT"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">kegg.gs</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## $`hsa00010_Glycolysis_/_Gluconeogenesis`</span>
<span class="co">##  [1] "A0A024QZ64" "A0A024R0Y5" "A0A024R145" "A0A024R5D8" "A0A024R713"</span>
<span class="co">##  [6] "A0A024R782" "A0A024R7R1" "A0A024RBX9" "A0A0A0MS74" "A0A0K0K1K1"</span>
<span class="co">## [11] "A0A140VJM9" "A0A140VJR3" "A0A140VKA7" "A8K7J7"     "B3KT70"    </span>
<span class="co">## [16] "B7Z6C2"     "B7Z9E5"     "B9EKV4"     "J3KNX1"     "O00757"    </span>
<span class="co">## [21] "P00325"     "P00326"     "P00338"     "P00558"     "P04075"    </span>
<span class="co">## [26] "P04406"     "P05091"     "P06733"     "P06744"     "P07195"    </span>
<span class="co">## [31] "P07327"     "P08319"     "P09104"     "P09467"     "P10515"    </span>
<span class="co">## [36] "P11177"     "P11766"     "P13929"     "P14550"     "P14618"    </span>
<span class="co">## [41] "P15259"     "P17858"     "P28332"     "P29803"     "P30613"    </span>
<span class="co">## [46] "P30837"     "P30838"     "P35557"     "P35558"     "P35575"    </span>
<span class="co">## [51] "P40394"     "P49419"     "P51648"     "P52789"     "P60174"    </span>
<span class="co">## [56] "Q01813"     "Q6ZMR3"     "Q8N0Y7"     "Q96C23"     "Q96FY7"    </span>
<span class="co">## [61] "Q96G03"     "Q9BRR6"     "Q9BUM1"     "Q9NQR9"     "Q9NUB1"    </span>
<span class="co">## [66] "Q9UNW1"    </span>
<span class="co">## </span>
<span class="co">## $`hsa00020_Citrate_cycle_(TCA_cycle)`</span>
<span class="co">##  [1] "A0A024R1T9" "A0A024R5C5" "A0A024R713" "A0A024RB75" "A0A024RBX9"</span>
<span class="co">##  [6] "A0A087X2E5" "A0A0A0MS74" "A0A0S2Z4C3" "A0A0S2Z4H7" "A0A140VJQ5"</span>
<span class="co">## [11] "B3KTM1"     "B7Z6J1"     "E5KS60"     "O75874"     "P10515"    </span>
<span class="co">## [16] "P11177"     "P21399"     "P21912"     "P29803"     "P31040"    </span>
<span class="co">## [21] "P35558"     "P40925"     "P48735"     "P50213"     "P51553"    </span>
<span class="co">## [26] "P53597"     "Q96I99"     "Q99643"     "Q99798"     "Q9ULD0"    </span>
<span class="co">## </span>
<span class="co">## $hsa00030_Pentose_phosphate_pathway</span>
<span class="co">##  [1] "A0A024QZ64" "A0A024R0Y5" "A0A024R145" "A0A0K0K1K7" "A0A140VK41"</span>
<span class="co">##  [6] "A0A140VK56" "A0A140VKC2" "B4E2U0"     "B7Z6C2"     "B7Z7I0"    </span>
<span class="co">## [11] "B7ZB02"     "E9PPM8"     "O00757"     "P04075"     "P06744"    </span>
<span class="co">## [16] "P09467"     "P11413"     "P17858"     "P21108"     "P29401"    </span>
<span class="co">## [21] "P49247"     "Q01813"     "Q15493"     "Q2QD12"     "Q5T6J7"    </span>
<span class="co">## [26] "Q8IVS8"     "Q96AT9"     "Q96G03"     "Q9H477"     "R4GMU1"    </span>
<span class="co">## </span>
<span class="co">## $hsa00040_Pentose_and_glucuronate_interconversions</span>
<span class="co">##  [1] "A0A024R7A8" "A0A140T9Z0" "A0A140VJM1" "A0A140VKE1" "O60218"    </span>
<span class="co">##  [6] "O60656"     "O60701"     "O75191"     "O75310"     "O75795"    </span>
<span class="co">## [11] "P06133"     "P08236"     "P14550"     "P16662"     "P19224"    </span>
<span class="co">## [16] "P22309"     "P22310"     "P35503"     "P35504"     "P36537"    </span>
<span class="co">## [21] "P54855"     "Q00796"     "Q2QD12"     "Q5DSZ6"     "Q5DSZ7"    </span>
<span class="co">## [26] "Q5DT02"     "Q6UWM9"     "Q7Z4W1"     "Q96AT9"     "Q96C11"    </span>
<span class="co">## [31] "Q9BY64"     "Q9UEF7"     "Q9UQ10"     "Q9Y2S2"     "Q9Y4X1"    </span>
<span class="co">## </span>
<span class="co">## $hsa00051_Fructose_and_mannose_metabolism</span>
<span class="co">##  [1] "A0A024QZ64" "A0A024R0Y5" "A0A024R145" "A0A024R1U5" "A0A024R482"</span>
<span class="co">##  [6] "A0A024R7A8" "A0A024R7R1" "A0A0S2Z4J6" "A0A140VJH7" "A0A140VJM6"</span>
<span class="co">## [11] "A8K7J7"     "B3KT70"     "B7Z8A0"     "H3BPP3"     "I1Z9G4"    </span>
<span class="co">## [16] "O00757"     "O14772"     "O60218"     "O60547"     "O60825"    </span>
<span class="co">## [21] "P04075"     "P09467"     "P17858"     "P52789"     "P60174"    </span>
<span class="co">## [26] "Q00796"     "Q01813"     "Q13630"     "Q16877"     "Q7L5Y1"    </span>
<span class="co">## [31] "Q8N0W3"     "Q9NQ88"     "Q9Y5P6"    </span>
<span class="co">## </span>
<span class="co">## $hsa00052_Galactose_metabolism</span>
<span class="co">##  [1] "A0A024R0Y5" "A0A024R7A8" "A0A024R7R1" "A0A080YV01" "A0A0S2Z3Y7"</span>
<span class="co">##  [6] "A0A140VKE1" "A8K7J7"     "B3KT70"     "B7Z6C2"     "H3BN99"    </span>
<span class="co">## [11] "O43451"     "O60218"     "O60909"     "P06280"     "P09848"    </span>
<span class="co">## [16] "P10253"     "P14410"     "P15291"     "P16278"     "P17858"    </span>
<span class="co">## [21] "P35557"     "P35575"     "P51570"     "P52789"     "Q01813"    </span>
<span class="co">## [26] "Q14376"     "Q2M2H8"     "Q96C23"     "Q96G03"     "Q9BUM1"    </span>
<span class="co">## [31] "Q9NQR9"</span></code></pre>
<div class="sourceCode" id="section-cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">go.gs</span> <span class="op">&lt;-</span> <span class="fu">EnrichmentBrowser</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/EnrichmentBrowser/man/getGenesets.html" class="external-link">getGenesets</a></span><span class="op">(</span>org <span class="op">=</span> <span class="st">"hsa"</span>, db <span class="op">=</span> <span class="st">"go"</span>, gene.id.type <span class="op">=</span> <span class="st">"UNIPROT"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span></code></pre>
<div class="sourceCode" id="section-cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">go.gs</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## $`GO:0000002_mitochondrial_genome_maintenance`</span>
<span class="co">##  [1] "A0A0S2Z3H3" "A0A0S2Z3Z9" "A0A0S2Z417" "B2RBL3"     "O60313"    </span>
<span class="co">##  [6] "P36776"     "Q96CQ1"     "Q9BQP7"     "Q9BSK2"     "Q9H611"    </span>
<span class="co">## [11] "Q9Y243"    </span>
<span class="co">## </span>
<span class="co">## $`GO:0000003_reproduction`</span>
<span class="co">## [1] "O43555" "O75900" "P01148" "Q5TKA1"</span>
<span class="co">## </span>
<span class="co">## $`GO:0000012_single_strand_break_repair`</span>
<span class="co">## [1] "A0A024R6L5" "A0A024RE06" "A8K128"     "B2RCY5"     "P09430"    </span>
<span class="co">## [6] "Q03468"     "Q13216"     "Q7Z2E3"     "Q8IW19"    </span>
<span class="co">## </span>
<span class="co">## $`GO:0000017_alpha-glucoside_transport`</span>
<span class="co">## [1] "P13866" "P31639"</span>
<span class="co">## </span>
<span class="co">## $`GO:0000018_regulation_of_DNA_recombination`</span>
<span class="co">## [1] "E9PB61" "P16871" "P52292" "P52294" "Q96FV9" "Q9H4L7"</span>
<span class="co">## </span>
<span class="co">## $`GO:0000019_regulation_of_mitotic_recombination`</span>
<span class="co">## [1] "A5D6Y3" "F8W7U8"</span></code></pre>
<p>Given the maximum scoring subnetwork and gene sets of interest, we
can now throw all kinds of set-based enrichment analysis (sbea) methods
or network-based enrichment analysis (nbea) methods onto the
problem.</p>
<div class="sourceCode" id="section-cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">EnrichmentBrowser</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/EnrichmentBrowser/man/sbea.html" class="external-link">sbeaMethods</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##  [1] "ora"        "safe"       "gsea"       "gsa"        "padog"     </span>
<span class="co">##  [6] "globaltest" "roast"      "camera"     "gsva"       "samgs"     </span>
<span class="co">## [11] "ebm"        "mgsa"</span></code></pre>
<div class="sourceCode" id="section-cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">EnrichmentBrowser</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/EnrichmentBrowser/man/nbea.html" class="external-link">nbeaMethods</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] "ggea"        "spia"        "pathnet"     "degraph"     "ganpa"      </span>
<span class="co">## [6] "cepa"        "topologygsa" "netgsa"      "neat"</span></code></pre>
<p>Those are conveniently wrapped in the EnrichmentBrowser package and
we are going to try out a couple of them here.</p>
<p>But for now we just take some information from the <a href="https://pubmed.ncbi.nlm.nih.gov/30526041/" class="external-link">paper</a> to illustrate
the concept:</p>
<p>NSUN2 is a nucleolus RNA m5C methyltransferase, regulator of cell
proliferation and cell differentiation, dysfunction associated with
intellectual diseases and cancer; reported to be a downstream target
gene of MYC oncogene, a master regulator of cell proliferation.</p>
<p>Among the pathways showing enrichment in the paper is Ribosome, so as
showcase let’s extract nodes participating in the gene set “Ribosome”
from the maximum scoring subnetwork.</p>
<div class="sourceCode" id="section-cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"Ribosome$"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">kegg.gs</span><span class="op">)</span><span class="op">)</span>
<span class="va">ribo</span> <span class="op">&lt;-</span> <span class="va">kegg.gs</span><span class="op">[[</span><span class="va">ind</span><span class="op">]</span><span class="op">]</span>
<span class="va">ribo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">ribo</span>, <span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/graphNEL-class.html" class="external-link">nodes</a></span><span class="op">(</span><span class="va">module</span><span class="op">)</span><span class="op">)</span>
<span class="va">ribo.gr</span> <span class="op">&lt;-</span> <span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/subGraph.html" class="external-link">subGraph</a></span><span class="op">(</span><span class="va">ribo</span>, <span class="va">module</span><span class="op">)</span>
<span class="va">ribo.gr</span></code></pre></div>
<pre><code><span class="co">## A graphNEL graph with directed edges</span>
<span class="co">## Number of Nodes = 26 </span>
<span class="co">## Number of Edges = 54</span></code></pre>
<p>Such a gene set induced sub-graph can then further explored in our
BioPlex node data and edge data shiny viewer.</p>
<p>For that we might be interested in mapping all kind of additional
data onto the nodes and the edges of the network.</p>
<p>For example, we might want to add the Bioplex proteome data onto the
graph for inspection via the <a href="https://github.com/ccb-hms/GraphViewer" class="external-link">shiny graph
viewer</a>.</p>
<div class="sourceCode" id="section-cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.prot</span> <span class="op">&lt;-</span> <span class="fu">BioPlex</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioPlex/man/getBioplexProteome.html" class="external-link">getBioplexProteome</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Using cached version from 2022-06-21 12:14:25</span></code></pre>
<div class="sourceCode" id="section-cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.prot</span></code></pre></div>
<pre><code><span class="co">## class: SummarizedExperiment </span>
<span class="co">## dim: 9604 10 </span>
<span class="co">## metadata(0):</span>
<span class="co">## assays(1): exprs</span>
<span class="co">## rownames(9604): P0CG40 Q8IXZ3-4 ... Q9H3S5 Q8WYQ3</span>
<span class="co">## rowData names(5): ENTREZID SYMBOL nr.peptides log2ratio adj.pvalue</span>
<span class="co">## colnames(10): HCT1 HCT2 ... HEK4 HEK5</span>
<span class="co">## colData names(1): cell.line</span></code></pre>
<div class="sourceCode" id="section-cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ribo.gr</span> <span class="op">&lt;-</span> <span class="fu">BioPlex</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioPlex/man/mapSummarizedExperimentOntoGraph.html" class="external-link">mapSummarizedExperimentOntoGraph</a></span><span class="op">(</span><span class="va">ribo.gr</span>, <span class="va">bp.prot</span>,
    col.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"HEK"</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, 
    rowdata.cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"nr.peptides"</span>, <span class="st">"log2ratio"</span>, <span class="st">"adj.pvalue"</span><span class="op">)</span>,
    prefix <span class="op">=</span> <span class="st">"bpprot_"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/nodeData-methods.html" class="external-link">nodeData</a></span><span class="op">(</span><span class="va">ribo.gr</span><span class="op">)</span>, n <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## $O75394</span>
<span class="co">## $O75394$ENTREZID</span>
<span class="co">## [1] "9553"</span>
<span class="co">## </span>
<span class="co">## $O75394$SYMBOL</span>
<span class="co">## [1] "MRPL33"</span>
<span class="co">## </span>
<span class="co">## $O75394$ISOFORM</span>
<span class="co">## [1] "O75394"</span>
<span class="co">## </span>
<span class="co">## $O75394$score</span>
<span class="co">## [1] 0.2549751</span>
<span class="co">## </span>
<span class="co">## $O75394$bpprot_HEK1</span>
<span class="co">## [1] 8.66958</span>
<span class="co">## </span>
<span class="co">## $O75394$bpprot_HEK2</span>
<span class="co">## [1] 8.36718</span>
<span class="co">## </span>
<span class="co">## $O75394$bpprot_HEK3</span>
<span class="co">## [1] 8.53282</span>
<span class="co">## </span>
<span class="co">## $O75394$bpprot_HEK4</span>
<span class="co">## [1] 8.6171</span>
<span class="co">## </span>
<span class="co">## $O75394$bpprot_HEK5</span>
<span class="co">## [1] 8.591</span>
<span class="co">## </span>
<span class="co">## $O75394$bpprot_nr.peptides</span>
<span class="co">## [1] 2</span>
<span class="co">## </span>
<span class="co">## $O75394$bpprot_log2ratio</span>
<span class="co">## [1] 0.4197192</span>
<span class="co">## </span>
<span class="co">## $O75394$bpprot_adj.pvalue</span>
<span class="co">## [1] 1.138452e-05</span>
<span class="co">## </span>
<span class="co">## </span>
<span class="co">## $P18124</span>
<span class="co">## $P18124$ENTREZID</span>
<span class="co">## [1] "6129"</span>
<span class="co">## </span>
<span class="co">## $P18124$SYMBOL</span>
<span class="co">## [1] "RPL7"</span>
<span class="co">## </span>
<span class="co">## $P18124$ISOFORM</span>
<span class="co">## [1] "P18124"</span>
<span class="co">## </span>
<span class="co">## $P18124$score</span>
<span class="co">## [1] 0.4136681</span>
<span class="co">## </span>
<span class="co">## $P18124$bpprot_HEK1</span>
<span class="co">## [1] 7.7563</span>
<span class="co">## </span>
<span class="co">## $P18124$bpprot_HEK2</span>
<span class="co">## [1] 6.93534</span>
<span class="co">## </span>
<span class="co">## $P18124$bpprot_HEK3</span>
<span class="co">## [1] 7.17503</span>
<span class="co">## </span>
<span class="co">## $P18124$bpprot_HEK4</span>
<span class="co">## [1] 7.73146</span>
<span class="co">## </span>
<span class="co">## $P18124$bpprot_HEK5</span>
<span class="co">## [1] 6.7052</span>
<span class="co">## </span>
<span class="co">## $P18124$bpprot_nr.peptides</span>
<span class="co">## [1] 4</span>
<span class="co">## </span>
<span class="co">## $P18124$bpprot_log2ratio</span>
<span class="co">## [1] 0.8111145</span>
<span class="co">## </span>
<span class="co">## $P18124$bpprot_adj.pvalue</span>
<span class="co">## [1] 4.220104e-06</span></code></pre>
</div>
<div class="section level3">
<h3 id="section-sessioninfo">SessionInfo<a class="anchor" aria-label="anchor" href="#section-sessioninfo"></a>
</h3>
<div class="sourceCode" id="section-cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## R version 4.2.0 (2022-04-22)</span>
<span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span>
<span class="co">## Running under: Ubuntu 20.04.4 LTS</span>
<span class="co">## </span>
<span class="co">## Matrix products: default</span>
<span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span>
<span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3</span>
<span class="co">## </span>
<span class="co">## locale:</span>
<span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span>
<span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span>
<span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span>
<span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span>
<span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span>
<span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span>
<span class="co">## </span>
<span class="co">## attached base packages:</span>
<span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span>
<span class="co">## [8] base     </span>
<span class="co">## </span>
<span class="co">## other attached packages:</span>
<span class="co">##  [1] org.Hs.eg.db_3.15.0         AnnotationDbi_1.58.0       </span>
<span class="co">##  [3] DESeq2_1.36.0               EnrichmentBrowser_2.26.0   </span>
<span class="co">##  [5] BioNet_1.56.0               RBGL_1.72.0                </span>
<span class="co">##  [7] graph_1.74.0                BioPlex_1.3.1              </span>
<span class="co">##  [9] SummarizedExperiment_1.26.1 Biobase_2.56.0             </span>
<span class="co">## [11] GenomicRanges_1.48.0        GenomeInfoDb_1.32.2        </span>
<span class="co">## [13] IRanges_2.30.0              S4Vectors_0.34.0           </span>
<span class="co">## [15] BiocGenerics_0.42.0         MatrixGenerics_1.8.0       </span>
<span class="co">## [17] matrixStats_0.62.0         </span>
<span class="co">## </span>
<span class="co">## loaded via a namespace (and not attached):</span>
<span class="co">##   [1] colorspace_2.0-3       ellipsis_0.3.2         rprojroot_2.0.3       </span>
<span class="co">##   [4] XVector_0.36.0         fs_1.5.2               bit64_4.0.5           </span>
<span class="co">##   [7] fansi_1.0.3            codetools_0.2-18       splines_4.2.0         </span>
<span class="co">##  [10] cachem_1.0.6           geneplotter_1.74.0     knitr_1.39            </span>
<span class="co">##  [13] jsonlite_1.8.0         annotate_1.74.0        GO.db_3.15.0          </span>
<span class="co">##  [16] dbplyr_2.2.0           png_0.1-7              shiny_1.7.1           </span>
<span class="co">##  [19] BiocManager_1.30.18    compiler_4.2.0         httr_1.4.3            </span>
<span class="co">##  [22] assertthat_0.2.1       Matrix_1.4-1           fastmap_1.1.0         </span>
<span class="co">##  [25] limma_3.52.2           cli_3.3.0              later_1.3.0           </span>
<span class="co">##  [28] htmltools_0.5.2        tools_4.2.0            igraph_1.3.2          </span>
<span class="co">##  [31] gtable_0.3.0           glue_1.6.2             GenomeInfoDbData_1.2.8</span>
<span class="co">##  [34] dplyr_1.0.9            rappdirs_0.3.3         Rcpp_1.0.8.3          </span>
<span class="co">##  [37] jquerylib_0.1.4        pkgdown_2.0.4          vctrs_0.4.1           </span>
<span class="co">##  [40] Biostrings_2.64.0      xfun_0.31              stringr_1.4.0         </span>
<span class="co">##  [43] mime_0.12              lifecycle_1.0.1        XML_3.99-0.10         </span>
<span class="co">##  [46] edgeR_3.38.1           zlibbioc_1.42.0        scales_1.2.0          </span>
<span class="co">##  [49] ragg_1.2.2             promises_1.2.0.1       parallel_4.2.0        </span>
<span class="co">##  [52] KEGGgraph_1.56.0       RColorBrewer_1.1-3     yaml_2.3.5            </span>
<span class="co">##  [55] curl_4.3.2             memoise_2.0.1          ggplot2_3.3.6         </span>
<span class="co">##  [58] sass_0.4.1             stringi_1.7.6          RSQLite_2.2.14        </span>
<span class="co">##  [61] highr_0.9              genefilter_1.78.0      desc_1.4.1            </span>
<span class="co">##  [64] BioPlexAnalysis_0.99.3 filelock_1.0.2         BiocParallel_1.30.3   </span>
<span class="co">##  [67] rlang_1.0.2            pkgconfig_2.0.3        systemfonts_1.0.4     </span>
<span class="co">##  [70] bitops_1.0-7           evaluate_0.15          lattice_0.20-45       </span>
<span class="co">##  [73] purrr_0.3.4            bit_4.0.4              tidyselect_1.1.2      </span>
<span class="co">##  [76] GSEABase_1.58.0        magrittr_2.0.3         R6_2.5.1              </span>
<span class="co">##  [79] generics_0.1.2         DelayedArray_0.22.0    DBI_1.1.3             </span>
<span class="co">##  [82] pillar_1.7.0           survival_3.3-1         KEGGREST_1.36.2       </span>
<span class="co">##  [85] RCurl_1.98-1.7         tibble_3.1.7           crayon_1.5.1          </span>
<span class="co">##  [88] utf8_1.2.2             BiocFileCache_2.4.0    rmarkdown_2.14        </span>
<span class="co">##  [91] locfit_1.5-9.5         grid_4.2.0             blob_1.2.3            </span>
<span class="co">##  [94] Rgraphviz_2.40.0       digest_0.6.29          xtable_1.8-4          </span>
<span class="co">##  [97] httpuv_1.6.5           textshaping_0.3.6      munsell_0.5.0         </span>
<span class="co">## [100] bslib_0.3.1</span></code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Ludwig Geistlinger, Robert Gentleman.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.4.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
