<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title> • BioPlexAnalysis</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="">
<meta property="og:description" content="BioPlexAnalysis">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioPlexAnalysis</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/BioNet.html">BioNet analysis</a>
</li>
<li>
  <a href="../articles/PFAM.html">PFAM analysis</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ccb-hms/BioPlexAnalysis">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="BioNet_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip></h1>
            
      
      
      <div class="hidden name"><code>BioNet.Rmd</code></div>

    </div>

    
    
<div id="setup" class="section level1">
<h1 class="hasAnchor">
<a href="#setup" class="anchor"></a>Setup</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">BioPlex</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bionet.bioapps.biozentrum.uni-wuerzburg.de/">BioNet</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">EnrichmentBrowser</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">graph</span><span class="op">)</span></code></pre></div>
</div>
<div id="bionet" class="section level1">
<h1 class="hasAnchor">
<a href="#bionet" class="anchor"></a>BioNet</h1>
<p>Get the latest version of the 293T PPI network:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.293t</span> <span class="op">&lt;-</span> <span class="fu">BioPlex</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioPlex/man/getBioPlex.html">getBioPlex</a></span><span class="op">(</span>cell.line <span class="op">=</span> <span class="st">"293T"</span>, version <span class="op">=</span> <span class="st">"3.0"</span><span class="op">)</span></code></pre></div>
<pre><code>## Using cached version from 2021-06-28 05:29:44</code></pre>
<p>and turn into a graph object:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.gr</span> <span class="op">&lt;-</span> <span class="fu">BioPlex</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioPlex/man/bioplex2graph.html">bioplex2graph</a></span><span class="op">(</span><span class="va">bp.293t</span><span class="op">)</span></code></pre></div>
<p>Get RNA-seq data for 293T cells from GEO: <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE122425">GSE122425</a></p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gse122425</span> <span class="op">&lt;-</span> <span class="fu">BioPlex</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioPlex/man/getGSE122425.html">getGSE122425</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## Using cached version from 2021-06-28 05:29:55</code></pre>
<p>Differential expression analysis between NSUN2 knockout and wild type:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gse122425</span><span class="op">$</span><span class="va">GROUP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span>, each <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="va">gse122425</span> <span class="op">&lt;-</span> <span class="fu">EnrichmentBrowser</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/EnrichmentBrowser/man/deAna.html">deAna</a></span><span class="op">(</span><span class="va">gse122425</span>, de.method <span class="op">=</span> <span class="st">"DESeq2"</span><span class="op">)</span></code></pre></div>
<pre><code>## Excluding 38949 genes not satisfying min.cpm threshold</code></pre>
<pre><code>## Corresponding software package not found: DESeq2
## Make sure that you have it installed.</code></pre>
<pre><code>## 'getOption("repos")' replaces Bioconductor standard repositories, see
## '?repositories' for details
## 
## replacement repositories:
##     CRAN: https://packagemanager.rstudio.com/all/__linux__/focal/latest</code></pre>
<pre><code>## Bioconductor version 3.13 (BiocManager 1.30.16), R 4.1.0 (2021-05-18)</code></pre>
<pre><code>## Installing package(s) 'DESeq2'</code></pre>
<pre><code>## also installing the dependencies 'formatR', 'lambda.r', 'futile.options', 'futile.logger', 'snow', 'BiocParallel', 'genefilter', 'geneplotter', 'RcppArmadillo'</code></pre>
<p>Inspect results:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">rowData</span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 18956 rows and 8 columns
##                        SYMBOL          KO                     GO    length
##                   &lt;character&gt; &lt;character&gt;            &lt;character&gt; &lt;integer&gt;
## ENSG00000227232        WASH7P      K18461                      _     15444
## ENSG00000228463    AP006222.2           _                      _     39639
## ENSG00000237094 RP4-669L17.10           _                      _    136229
## ENSG00000225630      MTND2P28      K03879                      _      1044
## ENSG00000237973  hsa-mir-6723      K02256                      _      1543
## ...                       ...         ...                    ...       ...
## ENSG00000185973         TMLHE      K00474 molecular_function:i..    179830
## ENSG00000124333         VAMP7      K08515 biological_process:a..     62478
## ENSG00000182484        WASH6P      K18461 molecular_function:m..      4885
## ENSG00000215414       PSMA6P1      K02730                      _       741
## ENSG00000185275        CD24P4      K06469                      _       243
##                        FC DESeq2.STAT        PVAL    ADJ.PVAL
##                 &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;   &lt;numeric&gt;
## ENSG00000227232  0.171520    0.724534    0.468738    0.648277
## ENSG00000228463  0.445551    1.194110    0.232435    0.419636
## ENSG00000237094 -0.318485   -0.446073    0.655545    0.791683
## ENSG00000225630 -0.404707   -1.227372    0.219683    0.404906
## ENSG00000237973 -0.337735   -0.983072    0.325572    0.518622
## ...                   ...         ...         ...         ...
## ENSG00000185973  0.111457    0.861636 3.88888e-01 5.79293e-01
## ENSG00000124333  3.430248    4.951910 7.34886e-07 2.15608e-05
## ENSG00000182484 -0.247212   -0.398845 6.90007e-01 8.17000e-01
## ENSG00000215414  0.543926    0.991335 3.21522e-01 5.14536e-01
## ENSG00000185275  0.452405    1.596938 1.10280e-01 2.59482e-01</code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">not.na</span> <span class="op">&lt;-</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span><span class="op">$</span><span class="va">PVAL</span><span class="op">)</span>
<span class="va">gse122425</span> <span class="op">&lt;-</span> <span class="va">gse122425</span><span class="op">[</span><span class="va">not.na</span>,<span class="op">]</span>
<span class="fu">EnrichmentBrowser</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/EnrichmentBrowser/man/plots.html">volcano</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span><span class="op">$</span><span class="va">FC</span>, <span class="fu">rowData</span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span><span class="op">$</span><span class="va">ADJ.PVAL</span><span class="op">)</span></code></pre></div>
<p><img src="BioNet_files/figure-html/de-results-gse122425-1.png" width="700"></p>
<p>Change rownames to gene symbols for intersection with bioplex graph:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">rowData</span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span><span class="op">$</span><span class="va">ENSEMBL</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">rowData</span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span><span class="op">$</span><span class="va">SYMBOL</span></code></pre></div>
<p>Intersect bioplex graph and expression dataset:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">syms</span> <span class="op">&lt;-</span> <span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/nodeData-methods.html">nodeData</a></span><span class="op">(</span><span class="va">bp.gr</span>, <span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/graphNEL-class.html">nodes</a></span><span class="op">(</span><span class="va">bp.gr</span><span class="op">)</span>, <span class="st">"SYMBOL"</span><span class="op">)</span>
<span class="va">syms</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span><span class="op">(</span><span class="va">syms</span><span class="op">)</span>
<span class="va">isect</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span><span class="op">(</span><span class="va">syms</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span><span class="op">)</span> 
<span class="va">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">isect</span>, <span class="va">syms</span><span class="op">)</span> 
<span class="va">bp.sgr</span> <span class="op">&lt;-</span> <span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/subGraph.html">subGraph</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">syms</span><span class="op">)</span><span class="op">[</span><span class="va">ind</span><span class="op">]</span>, <span class="va">bp.gr</span><span class="op">)</span>
<span class="va">bp.sgr</span> </code></pre></div>
<pre><code>## A graphNEL graph with directed edges
## Number of Nodes = 11015 
## Number of Edges = 79341</code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.sgr</span> <span class="op">&lt;-</span> <span class="fu">BioNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioNet/man/rmSelfLoops.html">rmSelfLoops</a></span><span class="op">(</span><span class="va">bp.sgr</span><span class="op">)</span>
<span class="va">bp.sgr</span></code></pre></div>
<pre><code>## A graphNEL graph with directed edges
## Number of Nodes = 11015 
## Number of Edges = 79341</code></pre>
<p>Get the DE p-values for each node:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">syms</span> <span class="op">&lt;-</span> <span class="va">syms</span><span class="op">[</span><span class="va">ind</span><span class="op">]</span>
<span class="va">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">isect</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span><span class="op">)</span>
<span class="va">pvals</span> <span class="op">&lt;-</span> <span class="fu">rowData</span><span class="op">(</span><span class="va">gse122425</span><span class="op">)</span><span class="op">[</span><span class="va">ind</span>, <span class="st">"PVAL"</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">syms</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">)</span></code></pre></div>
<pre><code>##     P00813     Q8N7W2     Q6ZMN8     P55039     Q9NZL4     Q6ICG6 
## 0.57137317 0.05761896 0.04112662 0.22823806 0.84595685 0.38653260</code></pre>
<p>To score each node of the network we fit a Beta-uniform mixture model (BUM) to the p-value distribution and subsequently use the parameters of the model for the scoring function:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fb</span> <span class="op">&lt;-</span> <span class="fu">BioNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioNet/man/fitBumModel.html">fitBumModel</a></span><span class="op">(</span><span class="va">pvals</span><span class="op">)</span></code></pre></div>
<p><img src="BioNet_files/figure-html/bum-bp-gse122425-1.png" width="700"></p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">scores</span> <span class="op">&lt;-</span> <span class="fu">BioNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioNet/man/scoreNodes.html">scoreNodes</a></span><span class="op">(</span><span class="va">bp.sgr</span>, <span class="va">fb</span>, fdr <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">scores</span><span class="op">)</span></code></pre></div>
<pre><code>##     P00813     Q8N7W2     Q6ZMN8     P55039     Q9NZL4     Q6ICG6 
## -2.3408846 -0.4505579 -0.1727211 -1.5847729 -2.6642287 -2.0188585</code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/summary.html">summary</a></span><span class="op">(</span><span class="va">scores</span><span class="op">)</span></code></pre></div>
<pre><code>##     Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
##  -2.8019  -2.1898  -1.2382   0.3586   0.6252 265.2463</code></pre>
<div id="first-reduction-step-unsupervised-detection-of-maximum-scoring-subnetwork" class="section level2">
<h2 class="hasAnchor">
<a href="#first-reduction-step-unsupervised-detection-of-maximum-scoring-subnetwork" class="anchor"></a>First reduction step: unsupervised detection of maximum scoring subnetwork</h2>
<p>Here we use a fast heuristic approach to calculate an approximation to theoptimal scoring subnetwork.<br>
An optimal solution can be calculated using the heinz algorithm requiring a commercial CPLEX license.</p>
<p>Note: this will run for a couple of minutes so it will be a good idea to cache the result to not recompute this over and over again.</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">module</span> <span class="op">&lt;-</span> <span class="fu">BioNet</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioNet/man/runFastHeinz.html">runFastHeinz</a></span><span class="op">(</span><span class="va">bp.sgr</span>, <span class="va">scores</span><span class="op">)</span>
<span class="va">module</span></code></pre></div>
<pre><code>## A graphNEL graph with directed edges
## Number of Nodes = 2851 
## Number of Edges = 8524</code></pre>
<p>For some reason <code><a href="https://rdrr.io/pkg/BioNet/man/runFastHeinz.html">BioNet::runFastHeinz</a></code> drops the <code>edgeData</code> of the input graph, so we need to re-annotate those to the module. There doesn’t seem to be a convenient way to bulk transfer <code>edgeData</code> from one graph to another, but rather only for one edge and one attribute at a time.</p>
<p>With a little abuse, we help ourselves via accessing slots directly.</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">en</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/edgeData-methods.html">edgeData</a></span><span class="op">(</span><span class="va">module</span><span class="op">)</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/graph/man/edgeDataDefaults-methods.html">edgeDataDefaults</a></span><span class="op">(</span><span class="va">module</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/graph/man/edgeDataDefaults-methods.html">edgeDataDefaults</a></span><span class="op">(</span><span class="va">bp.sgr</span><span class="op">)</span>
<span class="va">module</span><span class="op">@</span><span class="va">edgeData</span><span class="op">@</span><span class="va">data</span><span class="op">[</span><span class="va">en</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/graph/man/edgeData-methods.html">edgeData</a></span><span class="op">(</span><span class="va">bp.sgr</span><span class="op">)</span><span class="op">[</span><span class="va">en</span><span class="op">]</span></code></pre></div>
</div>
<div id="second-reduction-step-supervised-enrichment-analysis-of-maximum-scoring-subnetwork" class="section level2">
<h2 class="hasAnchor">
<a href="#second-reduction-step-supervised-enrichment-analysis-of-maximum-scoring-subnetwork" class="anchor"></a>Second reduction step: supervised enrichment analysis of maximum scoring subnetwork</h2>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">kegg.gs</span> <span class="op">&lt;-</span> <span class="fu">EnrichmentBrowser</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/EnrichmentBrowser/man/getGenesets.html">getGenesets</a></span><span class="op">(</span>org <span class="op">=</span> <span class="st">"hsa"</span>, db <span class="op">=</span> <span class="st">"kegg"</span>, gene.id.type <span class="op">=</span> <span class="st">"UNIPROT"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">kegg.gs</span><span class="op">)</span></code></pre></div>
<pre><code>## $`hsa00010_Glycolysis_/_Gluconeogenesis`
##  [1] "A0A024QZ64" "A0A024R0Y5" "A0A024R145" "A0A024R5D8" "A0A024R713"
##  [6] "A0A024R782" "A0A024R7R1" "A0A024RBX9" "A0A0A0MS74" "A0A0K0K1K1"
## [11] "A0A140VJM9" "A0A140VJR3" "A0A140VKA7" "A8K7J7"     "B3KT70"    
## [16] "B7Z6C2"     "B7Z9E5"     "B9EKV4"     "J3KNX1"     "O00757"    
## [21] "P00325"     "P00326"     "P00338"     "P00558"     "P04075"    
## [26] "P04406"     "P05091"     "P06733"     "P06744"     "P07195"    
## [31] "P07327"     "P08319"     "P09104"     "P09467"     "P10515"    
## [36] "P11177"     "P11766"     "P13929"     "P14550"     "P14618"    
## [41] "P15259"     "P17858"     "P28332"     "P29803"     "P30613"    
## [46] "P30837"     "P30838"     "P35557"     "P35558"     "P35575"    
## [51] "P40394"     "P49419"     "P51648"     "P52789"     "P60174"    
## [56] "Q01813"     "Q6ZMR3"     "Q8N0Y7"     "Q96C23"     "Q96FY7"    
## [61] "Q96G03"     "Q9BRR6"     "Q9BUM1"     "Q9NQR9"     "Q9NUB1"    
## [66] "Q9UNW1"    
## 
## $`hsa00020_Citrate_cycle_(TCA_cycle)`
##  [1] "A0A024R1T9" "A0A024R5C5" "A0A024R713" "A0A024RB75" "A0A024RBX9"
##  [6] "A0A087X2E5" "A0A0A0MS74" "A0A0S2Z4C3" "A0A0S2Z4H7" "A0A140VJQ5"
## [11] "B3KTM1"     "B7Z6J1"     "E5KS60"     "O75874"     "P10515"    
## [16] "P11177"     "P21399"     "P21912"     "P29803"     "P31040"    
## [21] "P35558"     "P40925"     "P48735"     "P50213"     "P51553"    
## [26] "P53597"     "Q96I99"     "Q99643"     "Q99798"     "Q9ULD0"    
## 
## $hsa00030_Pentose_phosphate_pathway
##  [1] "A0A024QZ64" "A0A024R0Y5" "A0A024R145" "A0A0K0K1K7" "A0A140VK41"
##  [6] "A0A140VK56" "A0A140VKC2" "B4E2U0"     "B7Z6C2"     "B7Z7I0"    
## [11] "B7ZB02"     "E9PPM8"     "O00757"     "P04075"     "P06744"    
## [16] "P09467"     "P11413"     "P17858"     "P21108"     "P29401"    
## [21] "P49247"     "Q01813"     "Q15493"     "Q2QD12"     "Q5T6J7"    
## [26] "Q8IVS8"     "Q96AT9"     "Q96G03"     "Q9H477"     "R4GMU1"    
## 
## $hsa00040_Pentose_and_glucuronate_interconversions
##  [1] "A0A024R7A8" "A0A140T9Z0" "A0A140VJM1" "A0A140VKE1" "O60218"    
##  [6] "O60656"     "O60701"     "O75191"     "O75310"     "O75795"    
## [11] "P06133"     "P08236"     "P14550"     "P16662"     "P19224"    
## [16] "P22309"     "P22310"     "P35503"     "P35504"     "P36537"    
## [21] "P54855"     "Q00796"     "Q2QD12"     "Q5DSZ6"     "Q5DSZ7"    
## [26] "Q5DT02"     "Q6UWM9"     "Q7Z4W1"     "Q96AT9"     "Q9BY64"    
## [31] "Q9UEF7"     "Q9UQ10"     "Q9Y2S2"     "Q9Y4X1"    
## 
## $hsa00051_Fructose_and_mannose_metabolism
##  [1] "A0A024QZ64" "A0A024R0Y5" "A0A024R145" "A0A024R1U5" "A0A024R482"
##  [6] "A0A024R7A8" "A0A024R7R1" "A0A0S2Z4J6" "A0A140VJH7" "A0A140VJM6"
## [11] "A8K7J7"     "B3KT70"     "B7Z8A0"     "H3BPP3"     "I1Z9G4"    
## [16] "O00757"     "O14772"     "O60218"     "O60547"     "O60825"    
## [21] "P04075"     "P09467"     "P17858"     "P52789"     "P60174"    
## [26] "Q00796"     "Q01813"     "Q13630"     "Q16877"     "Q7L5Y1"    
## [31] "Q8N0W3"     "Q9NQ88"     "Q9Y5P6"    
## 
## $hsa00052_Galactose_metabolism
##  [1] "A0A024R0Y5" "A0A024R7A8" "A0A024R7R1" "A0A080YV01" "A0A0S2Z3Y7"
##  [6] "A0A140VKE1" "A8K7J7"     "B3KT70"     "B7Z6C2"     "H3BN99"    
## [11] "O43451"     "O60218"     "O60909"     "P06280"     "P09848"    
## [16] "P10253"     "P14410"     "P15291"     "P16278"     "P17858"    
## [21] "P35557"     "P35575"     "P51570"     "P52789"     "Q01813"    
## [26] "Q14376"     "Q2M2H8"     "Q96C23"     "Q96G03"     "Q9BUM1"    
## [31] "Q9NQR9"</code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">go.gs</span> <span class="op">&lt;-</span> <span class="fu">EnrichmentBrowser</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/EnrichmentBrowser/man/getGenesets.html">getGenesets</a></span><span class="op">(</span>org <span class="op">=</span> <span class="st">"hsa"</span>, db <span class="op">=</span> <span class="st">"go"</span>, gene.id.type <span class="op">=</span> <span class="st">"UNIPROT"</span><span class="op">)</span></code></pre></div>
<pre><code>## </code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">go.gs</span><span class="op">)</span></code></pre></div>
<pre><code>## $`GO:0000002_mitochondrial_genome_maintenance`
##  [1] "A0A0S2Z3H3" "A0A0S2Z3Z9" "A0A0S2Z417" "B2RBL3"     "O60313"    
##  [6] "P36776"     "Q96CQ1"     "Q9BQP7"     "Q9BSK2"     "Q9H611"    
## [11] "Q9Y243"    
## 
## $`GO:0000003_reproduction`
## [1] "O43555" "O75900" "P01148" "Q5TKA1"
## 
## $`GO:0000012_single_strand_break_repair`
## [1] "A0A024R6L5" "A0A024RE06" "A8K128"     "B2RCY5"     "P09430"    
## [6] "Q03468"     "Q13216"     "Q7Z2E3"     "Q8IW19"    
## 
## $`GO:0000017_alpha-glucoside_transport`
## [1] "P13866" "P31639"
## 
## $`GO:0000018_regulation_of_DNA_recombination`
## [1] "E9PB61" "P16871" "P52292" "P52294" "Q96FV9" "Q9H4L7"
## 
## $`GO:0000019_regulation_of_mitotic_recombination`
## [1] "A5D6Y3" "F8W7U8"</code></pre>
<p>Given the maximum scoring subnetwork and gene sets of interest, we can now throw all kinds of set-based enrichment analysis (sbea) methods or network-based enrichment analysis (nbea) methods onto the problem.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">EnrichmentBrowser</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/EnrichmentBrowser/man/sbea.html">sbeaMethods</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>##  [1] "ora"        "safe"       "gsea"       "gsa"        "padog"     
##  [6] "globaltest" "roast"      "camera"     "gsva"       "samgs"     
## [11] "ebm"        "mgsa"</code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">EnrichmentBrowser</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/EnrichmentBrowser/man/nbea.html">nbeaMethods</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] "ggea"        "spia"        "pathnet"     "degraph"     "ganpa"      
## [6] "cepa"        "topologygsa" "netgsa"      "neat"</code></pre>
<p>Those are conveniently wrapped in the EnrichmentBrowser package and we are going to try out a couple of them here.</p>
<p>But for now we just take some information from the <a href="https://pubmed.ncbi.nlm.nih.gov/30526041/">paper</a> to illustrate the concept:</p>
<p>NSUN2 is a nucleolus RNA m5C methyltransferase, regulator of cell proliferation and cell differentiation, dysfunction associated with intellectual diseases and cancer; reported to be a downstream target gene of MYC oncogene, a master regulator of cell proliferation.</p>
<p>Among the pathways showing enrichment in the paper is Ribosome, so as showcase let’s extract nodes participating in the gene set “Ribosome” from the maximum scoring subnetwork.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"Ribosome$"</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">kegg.gs</span><span class="op">)</span><span class="op">)</span>
<span class="va">ribo</span> <span class="op">&lt;-</span> <span class="va">kegg.gs</span><span class="op">[[</span><span class="va">ind</span><span class="op">]</span><span class="op">]</span>
<span class="va">ribo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span><span class="op">(</span><span class="va">ribo</span>, <span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/graphNEL-class.html">nodes</a></span><span class="op">(</span><span class="va">module</span><span class="op">)</span><span class="op">)</span>
<span class="va">ribo.gr</span> <span class="op">&lt;-</span> <span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/subGraph.html">subGraph</a></span><span class="op">(</span><span class="va">ribo</span>, <span class="va">module</span><span class="op">)</span>
<span class="va">ribo.gr</span></code></pre></div>
<pre><code>## A graphNEL graph with directed edges
## Number of Nodes = 26 
## Number of Edges = 54</code></pre>
<p>Such a gene set induced sub-graph can then further explored in our BioPlex node data and edge data shiny viewer.</p>
<p>For that we might be interested in mapping all kind of additional data onto the nodes and the edges of the network.</p>
<p>For example, we might want to add the Bioplex proteome data onto the graph for inspection via the shiny graph viewer.</p>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.prot</span> <span class="op">&lt;-</span> <span class="fu">BioPlex</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioPlex/man/getBioplexProteome.html">getBioplexProteome</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## Using cached version from 2021-06-28 05:52:51</code></pre>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.prot</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 9604 10 
## metadata(0):
## assays(1): exprs
## rownames(9604): P0CG40 Q8IXZ3-4 ... Q9H3S5 Q8WYQ3
## rowData names(5): ENTREZID SYMBOL nr.peptides log2ratio adj.pvalue
## colnames(10): HCT1 HCT2 ... HEK4 HEK5
## colData names(1): cell.line</code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ribo.gr</span> <span class="op">&lt;-</span> <span class="fu">BioPlex</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioPlex/man/mapSummarizedExperimentOntoGraph.html">mapSummarizedExperimentOntoGraph</a></span><span class="op">(</span><span class="va">ribo.gr</span>, <span class="va">bp.prot</span>,
    col.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"HEK"</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">colnames</a></span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span>, value <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>, 
    rowdata.cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"nr.peptides"</span>, <span class="st">"log2ratio"</span>, <span class="st">"adj.pvalue"</span><span class="op">)</span>,
    prefix <span class="op">=</span> <span class="st">"bpprot_"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="fu">graph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/graph/man/nodeData-methods.html">nodeData</a></span><span class="op">(</span><span class="va">ribo.gr</span><span class="op">)</span>, n <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></code></pre></div>
<pre><code>## $O75394
## $O75394$ENTREZID
## [1] "9553"
## 
## $O75394$SYMBOL
## [1] "MRPL33"
## 
## $O75394$ISOFORM
## [1] "O75394"
## 
## $O75394$score
## [1] 0.2549766
## 
## $O75394$bpprot_HEK1
## [1] 8.66958
## 
## $O75394$bpprot_HEK2
## [1] 8.36718
## 
## $O75394$bpprot_HEK3
## [1] 8.53282
## 
## $O75394$bpprot_HEK4
## [1] 8.6171
## 
## $O75394$bpprot_HEK5
## [1] 8.591
## 
## $O75394$bpprot_nr.peptides
## [1] 2
## 
## $O75394$bpprot_log2ratio
## [1] 0.4197192
## 
## $O75394$bpprot_adj.pvalue
## [1] 1.13845e-05
## 
## 
## $P18124
## $P18124$ENTREZID
## [1] "6129"
## 
## $P18124$SYMBOL
## [1] "RPL7"
## 
## $P18124$ISOFORM
## [1] "P18124"
## 
## $P18124$score
## [1] 0.4136696
## 
## $P18124$bpprot_HEK1
## [1] 7.7563
## 
## $P18124$bpprot_HEK2
## [1] 6.93534
## 
## $P18124$bpprot_HEK3
## [1] 7.17503
## 
## $P18124$bpprot_HEK4
## [1] 7.73146
## 
## $P18124$bpprot_HEK5
## [1] 6.7052
## 
## $P18124$bpprot_nr.peptides
## [1] 4
## 
## $P18124$bpprot_log2ratio
## [1] 0.8111145
## 
## $P18124$bpprot_adj.pvalue
## [1] 4.2201e-06</code></pre>
</div>
<div id="sessioninfo" class="section level2">
<h2 class="hasAnchor">
<a href="#sessioninfo" class="anchor"></a>SessionInfo</h2>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.2 LTS
## 
## Matrix products: default
## BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
## 
## locale:
##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
## [8] methods   base     
## 
## other attached packages:
##  [1] org.Hs.eg.db_3.13.0         AnnotationDbi_1.54.1       
##  [3] DESeq2_1.32.0               EnrichmentBrowser_2.22.0   
##  [5] BioNet_1.52.0               RBGL_1.68.0                
##  [7] graph_1.70.0                BioPlex_0.99.1             
##  [9] SummarizedExperiment_1.22.0 Biobase_2.52.0             
## [11] GenomicRanges_1.44.0        GenomeInfoDb_1.28.0        
## [13] IRanges_2.26.0              S4Vectors_0.30.0           
## [15] BiocGenerics_0.38.0         MatrixGenerics_1.4.0       
## [17] matrixStats_0.59.0         
## 
## loaded via a namespace (and not attached):
##  [1] bitops_1.0-7           fs_1.5.0               bit64_4.0.5           
##  [4] RColorBrewer_1.1-2     filelock_1.0.2         httr_1.4.2            
##  [7] rprojroot_2.0.2        Rgraphviz_2.36.0       tools_4.1.0           
## [10] bslib_0.2.5.1          utf8_1.2.1             R6_2.5.0              
## [13] colorspace_2.0-2       DBI_1.1.1              withr_2.4.2           
## [16] tidyselect_1.1.1       bit_4.0.4              curl_4.3.2            
## [19] compiler_4.1.0         textshaping_0.3.5      desc_1.3.0            
## [22] DelayedArray_0.18.0    sass_0.4.0             scales_1.1.1          
## [25] KEGGgraph_1.52.0       genefilter_1.74.0      rappdirs_0.3.3        
## [28] pkgdown_1.6.1          systemfonts_1.0.2      stringr_1.4.0         
## [31] digest_0.6.27          rmarkdown_2.9          XVector_0.32.0        
## [34] pkgconfig_2.0.3        htmltools_0.5.1.1      highr_0.9             
## [37] dbplyr_2.1.1           fastmap_1.1.0          limma_3.48.1          
## [40] rlang_0.4.11           RSQLite_2.2.7          jquerylib_0.1.4       
## [43] generics_0.1.0         jsonlite_1.7.2         BiocParallel_1.26.0   
## [46] dplyr_1.0.7            RCurl_1.98-1.3         magrittr_2.0.1        
## [49] GO.db_3.13.0           GenomeInfoDbData_1.2.6 Matrix_1.3-3          
## [52] munsell_0.5.0          Rcpp_1.0.6             fansi_0.5.0           
## [55] lifecycle_1.0.0        stringi_1.6.2          yaml_2.2.1            
## [58] edgeR_3.34.0           zlibbioc_1.38.0        BiocFileCache_2.0.0   
## [61] grid_4.1.0             blob_1.2.1             crayon_1.4.1          
## [64] lattice_0.20-44        splines_4.1.0          Biostrings_2.60.1     
## [67] annotate_1.70.0        KEGGREST_1.32.0        locfit_1.5-9.4        
## [70] knitr_1.33             pillar_1.6.1           igraph_1.2.6          
## [73] geneplotter_1.70.0     XML_3.99-0.6           glue_1.4.2            
## [76] evaluate_0.14          BiocManager_1.30.16    png_0.1-7             
## [79] vctrs_0.3.8            gtable_0.3.0           purrr_0.3.4           
## [82] assertthat_0.2.1       ggplot2_3.3.4          cachem_1.0.5          
## [85] xfun_0.24              xtable_1.8-4           survival_3.2-11       
## [88] ragg_1.1.3             tibble_3.1.2           memoise_2.0.0         
## [91] ellipsis_0.3.2         GSEABase_1.54.0</code></pre>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Ludwig Geistlinger, Robert Gentleman.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
