<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title> • BioPlexAnalysis</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="">
<meta property="og:description" content="BioPlexAnalysis">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioPlexAnalysis</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.99.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/BioNet.html">BioNet analysis</a>
</li>
<li>
  <a href="../articles/PFAM.html">PFAM analysis</a>
</li>
<li>
  <a href="../articles/TranscriptomeProteome.html">Transcriptome-Proteome analysis</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ccb-hms/BioPlexAnalysis">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="TranscriptomeProteome_files/header-attrs-2.9/header-attrs.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip></h1>
            
      
      
      <div class="hidden name"><code>TranscriptomeProteome.Rmd</code></div>

    </div>

    
    
<div id="setup" class="section level1">
<h1 class="hasAnchor">
<a href="#setup" class="anchor"></a>Setup</h1>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ccb-hms/BioPlex">BioPlex</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/AnnotationDbi">AnnotationDbi</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">ExperimentHub</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">ExpressionAtlas</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/edgeR">edgeR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">graph</span><span class="op">)</span></code></pre></div>
</div>
<div id="ccle-transcriptome-and-proteome-data-for-hct116" class="section level1">
<h1 class="hasAnchor">
<a href="#ccle-transcriptome-and-proteome-data-for-hct116" class="anchor"></a>CCLE transcriptome and proteome data for HCT116</h1>
<p>Get CCLE transcriptome data for HCT116:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">atlasRes</span> <span class="op">&lt;-</span> <span class="fu">ExpressionAtlas</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExpressionAtlas/man/searchAtlasExperiments.html">searchAtlasExperiments</a></span><span class="op">(</span>
              properties <span class="op">=</span> <span class="st">"Cancer Cell Line Encyclopedia"</span>, 
              species <span class="op">=</span> <span class="st">"human"</span> <span class="op">)</span></code></pre></div>
<pre><code>## Searching for Expression Atlas experiments matching your query ...</code></pre>
<pre><code>## Query successful.</code></pre>
<pre><code>## Found 1 experiments matching your query.</code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">atlasRes</span></code></pre></div>
<pre><code>## DataFrame with 1 row and 4 columns
##     Accession      Species                  Type                  Title
##   &lt;character&gt;  &lt;character&gt;           &lt;character&gt;            &lt;character&gt;
## 1 E-MTAB-2770 Homo sapiens RNA-seq of coding RNA RNA-seq of 934 human..</code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ccle.trans</span> <span class="op">&lt;-</span> <span class="fu">ExpressionAtlas</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExpressionAtlas/man/getAtlasData.html">getAtlasData</a></span><span class="op">(</span><span class="va">atlasRes</span><span class="op">$</span><span class="va">Accession</span><span class="op">)</span></code></pre></div>
<pre><code>## Downloading Expression Atlas experiment summary from:
##  ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/atlas/experiments/E-MTAB-2770/E-MTAB-2770-atlasExperimentSummary.Rdata</code></pre>
<pre><code>## Successfully downloaded experiment summary object for E-MTAB-2770</code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ccle.trans</span> <span class="op">&lt;-</span> <span class="va">ccle.trans</span><span class="op">$</span><span class="va">`E-MTAB-2770`</span><span class="op">$</span><span class="va">rnaseq</span>
<span class="va">ccle.trans</span> <span class="op">&lt;-</span> <span class="va">ccle.trans</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"HCT 116"</span>, <span class="va">ccle.trans</span><span class="op">$</span><span class="va">cell_line</span><span class="op">)</span><span class="op">]</span>
<span class="va">ccle.trans</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 65217 1 
## metadata(4): pipeline filtering mapping quantification
## assays(1): counts
## rownames(65217): ENSG00000000003 ENSG00000000005 ... ENSG00000281921
##   ENSG00000281922
## rowData names(0):
## colnames(1): run_HCT_116.2
## colData names(4): AtlasAssayGroup organism cell_line disease</code></pre>
<p>Get the CCLE proteome data for HCT116:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu">ExperimentHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## snapshotDate(): 2021-05-18</code></pre>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ehdb</span> <span class="op">&lt;-</span> <span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">query</a></span><span class="op">(</span><span class="va">eh</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"gygi"</span>, <span class="st">"depmap"</span><span class="op">)</span><span class="op">)</span>
<span class="va">ccle.prot</span> <span class="op">&lt;-</span> <span class="va">ehdb</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">ehdb</span><span class="op">)</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<pre><code>## 'getOption("repos")' replaces Bioconductor standard repositories, see
## '?repositories' for details
## 
## replacement repositories:
##     CRAN: https://packagemanager.rstudio.com/all/__linux__/focal/latest</code></pre>
<pre><code>## Bioconductor version 3.13 (BiocManager 1.30.16), R 4.1.0 (2021-05-18)</code></pre>
<pre><code>## Installing package(s) 'depmap'</code></pre>
<pre><code>## Old packages: 'gargle', 'googledrive', 'RPostgres', 'testthat', 'vroom',
##   'Matrix', 'mgcv'</code></pre>
<pre><code>## snapshotDate(): 2021-05-18</code></pre>
<pre><code>## see ?depmap and browseVignettes('depmap') for documentation</code></pre>
<pre><code>## loading from cache</code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ccle.prot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="va">ccle.prot</span><span class="op">)</span>
<span class="va">ccle.prot</span> <span class="op">&lt;-</span> <span class="fu">BioPlex</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioPlex/man/ccleProteome2SummarizedExperiment.html">ccleProteome2SummarizedExperiment</a></span><span class="op">(</span><span class="va">ccle.prot</span><span class="op">)</span>
<span class="va">ccle.prot</span></code></pre></div>
<pre><code>## class: SummarizedExperiment 
## dim: 12755 1 
## metadata(0):
## assays(1): expr
## rownames(12755): P55011 P35453 ... Q99735 Q9P003
## rowData names(2): SYMBOL ENTREZID
## colnames(1): HCT116
## colData names(0):</code></pre>
<p>Connect to <a href="http://bioconductor.org/packages/AnnotationHub">AnnotationHub</a> and obtain OrgDb package for human:</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">ahdb</span> <span class="op">&lt;-</span> <span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"orgDb"</span>, <span class="st">"Homo sapiens"</span><span class="op">)</span><span class="op">)</span>
<span class="va">orgdb</span> <span class="op">&lt;-</span> <span class="va">ahdb</span><span class="op">[[</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">ahdb</span><span class="op">)</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p>Map to ENSEMBL for comparison with CCLE transcriptome data for HCT116:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rnames</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">mapIds</a></span><span class="op">(</span><span class="va">orgdb</span>,
                      keytype <span class="op">=</span> <span class="st">"UNIPROT"</span>,
                      column <span class="op">=</span> <span class="st">"ENSEMBL"</span>,
                      keys <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">ccle.prot</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<p>Subset to the ENSEMBL IDs that both datasets have in common</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">isect</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span><span class="op">(</span><span class="va">rnames</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">ccle.trans</span><span class="op">)</span><span class="op">)</span>
<span class="va">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">isect</span>, <span class="va">rnames</span><span class="op">)</span></code></pre></div>
<p>This should be rather RPKM, provided gene length from EDASeq:</p>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">assay</span><span class="op">(</span><span class="va">ccle.trans</span>, <span class="st">"cpm"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html">cpm</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">ccle.trans</span><span class="op">)</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>A look at general correlation between transcriptome and proteome:</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/cor.test.html">cor.test</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">ccle.trans</span>, <span class="st">"cpm"</span><span class="op">)</span><span class="op">[</span><span class="va">isect</span>,<span class="op">]</span>, 
         <span class="fu">assay</span><span class="op">(</span><span class="va">ccle.prot</span><span class="op">)</span><span class="op">[</span><span class="va">ind</span>,<span class="op">]</span>,
         use <span class="op">=</span> <span class="st">"complete.obs"</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##  Pearson's product-moment correlation
## 
## data:  assay(ccle.trans, "cpm")[isect, ] and assay(ccle.prot)[ind, ]
## t = 32.224, df = 7848, p-value &lt; 2.2e-16
## alternative hypothesis: true correlation is not equal to 0
## 95 percent confidence interval:
##  0.3221509 0.3612272
## sample estimates:
##       cor 
## 0.3418368</code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>trans <span class="op">=</span> <span class="fu">assay</span><span class="op">(</span><span class="va">ccle.trans</span>, <span class="st">"cpm"</span><span class="op">)</span><span class="op">[</span><span class="va">isect</span>,<span class="op">]</span>,
                 prot <span class="op">=</span> <span class="fu">assay</span><span class="op">(</span><span class="va">ccle.prot</span><span class="op">)</span><span class="op">[</span><span class="va">ind</span>,<span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">trans</span>, y <span class="op">=</span> <span class="va">prot</span><span class="op">)</span> <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bin_2d.html">geom_bin2d</a></span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">70</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_continuous.html">scale_fill_continuous</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"viridis"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"log2 CPM"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"log2 intensity"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: Removed 2690 rows containing non-finite values (stat_bin2d).</code></pre>
<p><img src="TranscriptomeProteome_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<p>Let’s check whether this looks very different when accounting for gene length. We therefore obtain gene length for the hg38 genome assembly (used for CCLE).</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ahdb</span> <span class="op">&lt;-</span> <span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TxDb"</span>, <span class="st">"Homo sapiens"</span><span class="op">)</span><span class="op">)</span>
<span class="va">txdb</span> <span class="op">&lt;-</span> <span class="va">ahdb</span><span class="op">[[</span><span class="st">"AH92591"</span><span class="op">]</span><span class="op">]</span>
<span class="va">gs</span> <span class="op">&lt;-</span> <span class="fu">GenomicFeatures</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html">genes</a></span><span class="op">(</span><span class="va">txdb</span><span class="op">)</span>
<span class="va">gs</span></code></pre></div>
<pre><code>## GRanges object with 27113 ranges and 1 metadata column:
##         seqnames              ranges strand |     gene_id
##            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;
##       1    chr19   58345178-58362751      - |           1
##      10     chr8   18391282-18401218      + |          10
##     100    chr20   44619522-44652233      - |         100
##    1000    chr18   27932879-28177946      - |        1000
##   10000     chr1 243488233-243851079      - |       10000
##     ...      ...                 ...    ... .         ...
##    9991     chr9 112217716-112333664      - |        9991
##    9992    chr21   34364006-34371381      + |        9992
##    9993    chr22   19036282-19122454      - |        9993
##    9994     chr6   89829894-89874436      + |        9994
##    9997    chr22   50523568-50526461      - |        9997
##   -------
##   seqinfo: 595 sequences (1 circular) from hg38 genome</code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">len</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu">width</span><span class="op">(</span><span class="va">gs</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">len</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span><span class="op">(</span><span class="va">gs</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">len</span><span class="op">)</span></code></pre></div>
<pre><code>##         1        10       100      1000     10000 100009613 
##     17574      9937     32712    245068    362847      3000</code></pre>
<p>This requires to map from Entrez IDs present for the gene length data to ENSEMBL IDs present in the transcriptomic data.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">eids</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html">mapIds</a></span><span class="op">(</span><span class="va">orgdb</span>,
                              column <span class="op">=</span> <span class="st">"ENTREZID"</span>,
                              keytype <span class="op">=</span> <span class="st">"ENSEMBL"</span>, 
                              keys <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">ccle.trans</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>## 'select()' returned 1:many mapping between keys and columns</code></pre>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">rowData</span><span class="op">(</span><span class="va">ccle.trans</span><span class="op">)</span><span class="op">$</span><span class="va">length</span> <span class="op">&lt;-</span> <span class="va">len</span><span class="op">[</span><span class="va">eids</span><span class="op">]</span></code></pre></div>
<p>We can now compute RPKM given the obtained gene lengths as input.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">assay</span><span class="op">(</span><span class="va">ccle.trans</span>, <span class="st">"rpkm"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html">rpkm</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">ccle.trans</span><span class="op">)</span>, 
                                         gene.length <span class="op">=</span> <span class="fu">rowData</span><span class="op">(</span><span class="va">ccle.trans</span><span class="op">)</span><span class="op">$</span><span class="va">length</span>,
                                         log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> </code></pre></div>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/cor.test.html">cor.test</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">ccle.trans</span>, <span class="st">"rpkm"</span><span class="op">)</span><span class="op">[</span><span class="va">isect</span>,<span class="op">]</span>,
         <span class="fu">assay</span><span class="op">(</span><span class="va">ccle.prot</span><span class="op">)</span><span class="op">[</span><span class="va">ind</span>,<span class="op">]</span>,
         use <span class="op">=</span> <span class="st">"complete.obs"</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##  Pearson's product-moment correlation
## 
## data:  assay(ccle.trans, "rpkm")[isect, ] and assay(ccle.prot)[ind, ]
## t = 29.79, df = 7814, p-value &lt; 2.2e-16
## alternative hypothesis: true correlation is not equal to 0
## 95 percent confidence interval:
##  0.2993008 0.3391210
## sample estimates:
##       cor 
## 0.3193519</code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>trans <span class="op">=</span> <span class="fu">assay</span><span class="op">(</span><span class="va">ccle.trans</span>, <span class="st">"rpkm"</span><span class="op">)</span><span class="op">[</span><span class="va">isect</span>,<span class="op">]</span>,
                 prot <span class="op">=</span> <span class="fu">assay</span><span class="op">(</span><span class="va">ccle.prot</span><span class="op">)</span><span class="op">[</span><span class="va">ind</span>,<span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">trans</span>, y <span class="op">=</span> <span class="va">prot</span><span class="op">)</span> <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bin_2d.html">geom_bin2d</a></span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">70</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_continuous.html">scale_fill_continuous</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"viridis"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"log2 RPKM"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"log2 intensity"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## Warning: Removed 2724 rows containing non-finite values (stat_bin2d).</code></pre>
<p><img src="TranscriptomeProteome_files/figure-html/unnamed-chunk-11-1.png" width="700"> –&gt;</p>
</div>
<div id="de-analysis-hek293-vs--hct116-transcriptomic-and-proteomic-level" class="section level1">
<h1 class="hasAnchor">
<a href="#de-analysis-hek293-vs--hct116-transcriptomic-and-proteomic-level" class="anchor"></a>DE analysis HEK293 vs. HCT116 (transcriptomic and proteomic level)</h1>
<p>Pull the HEK293 data:</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gse.293t</span> <span class="op">&lt;-</span> <span class="fu">BioPlex</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioPlex/man/getGSE122425.html">getGSE122425</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## Using cached version from 2021-07-12 05:34:43</code></pre>
<p>Pull the HCT116 data:</p>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">klijn</span> <span class="op">&lt;-</span> <span class="fu">ExpressionAtlas</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExpressionAtlas/man/getAtlasData.html">getAtlasData</a></span><span class="op">(</span><span class="st">"E-MTAB-2706"</span><span class="op">)</span></code></pre></div>
<pre><code>## Downloading Expression Atlas experiment summary from:
##  ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/atlas/experiments/E-MTAB-2706/E-MTAB-2706-atlasExperimentSummary.Rdata</code></pre>
<pre><code>## Successfully downloaded experiment summary object for E-MTAB-2706</code></pre>
<div class="sourceCode" id="cb51"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">klijn</span> <span class="op">&lt;-</span> <span class="va">klijn</span><span class="op">$</span><span class="va">`E-MTAB-2706`</span><span class="op">$</span><span class="va">rnaseq</span>
<span class="va">klijn</span></code></pre></div>
<pre><code>## class: RangedSummarizedExperiment 
## dim: 65217 622 
## metadata(4): pipeline filtering mapping quantification
## assays(1): counts
## rownames(65217): ENSG00000000003 ENSG00000000005 ... ENSG00000281921
##   ENSG00000281922
## rowData names(0):
## colnames(622): ERR413347 ERR413348 ... ERR414020 ERR415514
## colData names(12): AtlasAssayGroup organism ... media freeze_media</code></pre>
<p>Combine the both HCT116 samples:</p>
<div class="sourceCode" id="cb53"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ind2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html">grep</a></span><span class="op">(</span><span class="st">"HCT 116"</span>, <span class="va">klijn</span><span class="op">$</span><span class="va">cell_line</span><span class="op">)</span>
<span class="va">emat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cbind.html">cbind</a></span><span class="op">(</span><span class="fu">assay</span><span class="op">(</span><span class="va">ccle.trans</span><span class="op">)</span>, <span class="fu">assay</span><span class="op">(</span><span class="va">klijn</span><span class="op">)</span><span class="op">[</span>,<span class="va">ind2</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html">colnames</a></span><span class="op">(</span><span class="va">emat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"ccle"</span>, <span class="st">"klijn"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span><span class="op">(</span><span class="va">emat</span><span class="op">)</span></code></pre></div>
<pre><code>##                 ccle klijn
## ENSG00000000003 2471  1876
## ENSG00000000005    0     0
## ENSG00000000419 4082  3731
## ENSG00000000457 1160   676
## ENSG00000000460 1785  1206
## ENSG00000000938    0     0</code></pre>
<p>Combine with the HEK293 wildtype samples:</p>
<div class="sourceCode" id="cb55"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">isect</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">emat</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">gse.293t</span><span class="op">)</span><span class="op">)</span>
<span class="va">emat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cbind.html">cbind</a></span><span class="op">(</span><span class="va">emat</span><span class="op">[</span><span class="va">isect</span>,<span class="op">]</span>, <span class="fu">assay</span><span class="op">(</span><span class="va">gse.293t</span><span class="op">)</span><span class="op">[</span><span class="va">isect</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html">colnames</a></span><span class="op">(</span><span class="va">emat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"HCT"</span>, <span class="st">"HEK"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> </code></pre></div>
<p>Compute logCPMs to bring samples from different cell lines and experiments on the same scale using the limma-trend approach:</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dge</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html">DGEList</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">emat</span><span class="op">)</span>
<span class="va">dge</span><span class="op">$</span><span class="va">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"HCT"</span>, <span class="st">"HEK"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span>
<span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">dge</span><span class="op">$</span><span class="va">group</span><span class="op">)</span>
<span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html">filterByExpr</a></span><span class="op">(</span><span class="va">dge</span>, <span class="va">design</span><span class="op">)</span>
<span class="va">dge</span> <span class="op">&lt;-</span> <span class="va">dge</span><span class="op">[</span><span class="va">keep</span>,,keep.lib.sizes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span><span class="op">(</span><span class="va">dge</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 19484     5</code></pre>
<div class="sourceCode" id="cb58"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dge</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html">calcNormFactors</a></span><span class="op">(</span><span class="va">dge</span><span class="op">)</span>
<span class="va">logCPM</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html">cpm</a></span><span class="op">(</span><span class="va">dge</span>, log <span class="op">=</span> <span class="cn">TRUE</span>, prior.count <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html">lmFit</a></span><span class="op">(</span><span class="va">logCPM</span>, <span class="va">design</span><span class="op">)</span>
<span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html">eBayes</a></span><span class="op">(</span><span class="va">fit</span>, trend <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html">ncol</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code>##                      logFC  AveExpr          t      P.Value    adj.P.Val
## ENSG00000198786 -15.770778 2.377802 -101.69268 8.130301e-11 7.127842e-07
## ENSG00000176788   9.924371 4.197113   93.46125 1.338881e-10 7.127842e-07
## ENSG00000100219  11.372134 2.892771   89.94957 1.678907e-10 7.127842e-07
## ENSG00000227063  11.604079 3.031938   87.85518 1.929682e-10 7.127842e-07
## ENSG00000134871   9.388458 4.587640   87.50795 1.975368e-10 7.127842e-07
## ENSG00000261409  12.847018 3.975241   85.63962 2.244035e-10 7.127842e-07
## ENSG00000138829   9.497422 4.565920   83.74690 2.560814e-10 7.127842e-07
## ENSG00000133124  12.785690 3.928053   81.60748 2.983876e-10 7.267230e-07
## ENSG00000159217   8.133320 4.816707   78.01356 3.893508e-10 7.395928e-07
## ENSG00000041982   8.161008 4.436087   76.99581 4.207589e-10 7.395928e-07
##                        B
## ENSG00000198786 13.21201
## ENSG00000176788 13.05849
## ENSG00000100219 12.98240
## ENSG00000227063 12.93350
## ENSG00000134871 12.92513
## ENSG00000261409 12.87866
## ENSG00000138829 12.82908
## ENSG00000133124 12.76978
## ENSG00000159217 12.66162
## ENSG00000041982 12.62889</code></pre>
<div class="sourceCode" id="cb60"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tt</span> <span class="op">&lt;-</span> <span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html">topTable</a></span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html">ncol</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span>, number <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html">nrow</a></span><span class="op">(</span><span class="va">logCPM</span><span class="op">)</span>, sort.by <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></code></pre></div>
<p>Now let’s pull the BioPlex3 proteome data:</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.prot</span> <span class="op">&lt;-</span> <span class="fu">BioPlex</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioPlex/man/getBioplexProteome.html">getBioplexProteome</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code>## Using cached version from 2021-07-12 05:56:37</code></pre>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">rowData</span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span></code></pre></div>
<pre><code>## DataFrame with 9604 rows and 5 columns
##             ENTREZID      SYMBOL nr.peptides log2ratio  adj.pvalue
##          &lt;character&gt; &lt;character&gt;   &lt;integer&gt; &lt;numeric&gt;   &lt;numeric&gt;
## P0CG40     100131390         SP9           1 -2.819071 6.66209e-08
## Q8IXZ3-4      221833         SP8           3 -3.419888 6.94973e-07
## P55011          6558     SLC12A2           4  0.612380 4.85602e-06
## O60341         23028       KDM1A           7 -0.319695 5.08667e-04
## O14654          8471        IRS4           4 -5.951096 1.45902e-06
## ...              ...         ...         ...       ...         ...
## Q9H6X4         80194     TMEM134           2 -0.379342 7.67195e-05
## Q9BS91         55032     SLC35A5           1 -2.237634 8.75523e-05
## Q9UKJ5         26511       CHIC2           1 -0.614932 1.78756e-03
## Q9H3S5         93183        PIGM           1 -1.011397 8.91589e-06
## Q8WYQ3        400916     CHCHD10           1  0.743852 1.17163e-03</code></pre>
<p>Compare differential expression results on transcriptomic and proteomic level based on gene symbols as those are readily available:</p>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">isect</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span><span class="op">(</span><span class="fu">rowData</span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span><span class="op">$</span><span class="va">SYMBOL</span>, 
                   <span class="fu">rowData</span><span class="op">(</span><span class="va">gse.293t</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">logCPM</span><span class="op">)</span>, <span class="st">"SYMBOL"</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">isect</span><span class="op">)</span></code></pre></div>
<pre><code>## [1] 8980</code></pre>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ind.trans</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">isect</span>, <span class="fu">rowData</span><span class="op">(</span><span class="va">gse.293t</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span><span class="op">(</span><span class="va">logCPM</span><span class="op">)</span>, <span class="st">"SYMBOL"</span><span class="op">]</span><span class="op">)</span>
<span class="va">ind.prot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html">match</a></span><span class="op">(</span><span class="va">isect</span>, <span class="fu">rowData</span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span><span class="op">$</span><span class="va">SYMBOL</span><span class="op">)</span></code></pre></div>
<p>We need to switch here the sign of the fold change because the transcriptome is HEK-vs-HCT, the proteome is HCT-vs-HEK:</p>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/cor.test.html">cor.test</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span> <span class="op">*</span> <span class="va">tt</span><span class="op">[</span><span class="va">ind.trans</span>, <span class="st">"logFC"</span><span class="op">]</span>, 
         <span class="fu">rowData</span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span><span class="op">[</span><span class="va">ind.prot</span>, <span class="st">"log2ratio"</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code>## 
##  Pearson's product-moment correlation
## 
## data:  -1 * tt[ind.trans, "logFC"] and rowData(bp.prot)[ind.prot, "log2ratio"]
## t = 71.548, df = 8978, p-value &lt; 2.2e-16
## alternative hypothesis: true correlation is not equal to 0
## 95 percent confidence interval:
##  0.5892634 0.6156127
## sample estimates:
##       cor 
## 0.6026023</code></pre>
<div class="sourceCode" id="cb70"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>trans <span class="op">=</span> <span class="op">-</span><span class="fl">1</span> <span class="op">*</span> <span class="va">tt</span><span class="op">[</span><span class="va">ind.trans</span>, <span class="st">"logFC"</span><span class="op">]</span>,
                 prot <span class="op">=</span> <span class="fu">rowData</span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span><span class="op">[</span><span class="va">ind.prot</span>, <span class="st">"log2ratio"</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">trans</span>, y <span class="op">=</span> <span class="va">prot</span><span class="op">)</span> <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bin_2d.html">geom_bin2d</a></span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">70</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_continuous.html">scale_fill_continuous</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"viridis"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span><span class="op">(</span><span class="st">"log2FC (transcriptome)"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span><span class="op">(</span><span class="st">"log2FC (proteome)"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="TranscriptomeProteome_files/figure-html/unnamed-chunk-22-1.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Ludwig Geistlinger, Robert Gentleman.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
