<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title> • BioPlexAnalysis</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="">
<meta property="og:description" content="BioPlexAnalysis">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioPlexAnalysis</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Vignettes
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BioNet.html">Maximum scoring subnetwork analysis</a>
    </li>
    <li>
      <a href="../articles/CORUM.html">Protein complex overlap analysis</a>
    </li>
    <li>
      <a href="../articles/PDB.html">Mapping PPIs onto 3D protein structures</a>
    </li>
    <li>
      <a href="../articles/PFAM.html">Protein domain association analysis</a>
    </li>
    <li>
      <a href="../articles/TranscriptomeProteome.html">Transcriptome and Proteome analysis</a>
    </li>
    <li>
      <a href="../articles/GraphFrames.html">GraphFrames backend</a>
    </li>
  </ul>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/ccb-hms/BioPlexAnalysis" class="external-link">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip></h1>
            
      
      
      <div class="hidden name"><code>TranscriptomeProteome.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ccb-hms/BioPlex" class="external-link">BioPlex</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/AnnotationDbi" class="external-link">AnnotationDbi</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">AnnotationHub</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ExperimentHub</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">ExpressionAtlas</span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicFeatures" class="external-link">GenomicFeatures</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/edgeR" class="external-link">edgeR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://bioinf.wehi.edu.au/limma" class="external-link">limma</a></span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="ccle-transcriptome-and-proteome-data-for-hct116">CCLE transcriptome and proteome data for HCT116<a class="anchor" aria-label="anchor" href="#ccle-transcriptome-and-proteome-data-for-hct116"></a>
</h2>
<p>Get CCLE transcriptome data for HCT116:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">atlasRes</span> <span class="op">&lt;-</span> <span class="fu">ExpressionAtlas</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExpressionAtlas/man/searchAtlasExperiments.html" class="external-link">searchAtlasExperiments</a></span><span class="op">(</span>
              properties <span class="op">=</span> <span class="st">"Cancer Cell Line Encyclopedia"</span>, 
              species <span class="op">=</span> <span class="st">"human"</span> <span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Searching for Expression Atlas experiments matching your query ...</span></code></pre>
<pre><code><span class="co">## Query successful.</span></code></pre>
<pre><code><span class="co">## No results found. Cannot continue.</span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">atlasRes</span></code></pre></div>
<pre><code><span class="co">## NULL</span></code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ccle.trans</span> <span class="op">&lt;-</span> <span class="fu">ExpressionAtlas</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExpressionAtlas/man/getAtlasExperiment.html" class="external-link">getAtlasExperiment</a></span><span class="op">(</span><span class="st">"E-MTAB-2770"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Downloading Expression Atlas experiment summary from:</span>
<span class="co">##  ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/atlas/experiments/E-MTAB-2770/E-MTAB-2770-atlasExperimentSummary.Rdata</span></code></pre>
<pre><code><span class="co">## Successfully downloaded experiment summary object for E-MTAB-2770</span></code></pre>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ccle.trans</span> <span class="op">&lt;-</span> <span class="va">ccle.trans</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="va">ccle.trans</span> <span class="op">&lt;-</span> <span class="va">ccle.trans</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"HCT 116"</span>, <span class="va">ccle.trans</span><span class="op">$</span><span class="va">cell_line</span><span class="op">)</span><span class="op">]</span>
<span class="va">ccle.trans</span></code></pre></div>
<pre><code><span class="co">## class: RangedSummarizedExperiment </span>
<span class="co">## dim: 58735 0 </span>
<span class="co">## metadata(4): pipeline filtering mapping quantification</span>
<span class="co">## assays(1): counts</span>
<span class="co">## rownames(58735): ENSG00000000003 ENSG00000000005 ... ENSG00000285993</span>
<span class="co">##   ENSG00000285994</span>
<span class="co">## rowData names(0):</span>
<span class="co">## colnames(0):</span>
<span class="co">## colData names(4): AtlasAssayGroup organism cell_line disease</span></code></pre>
<p>There is currently an <a href="https://github.com/ebi-gene-expression-group/bioconductor-ExpressionAtlas/issues/6" class="external-link">issue
with obtaining E-MTAB-2770</a> via the <a href="https://bioconductor.org/packages/ExpressionAtlas" class="external-link">ExpressionAtlas</a>
package. We therefore pull the file directly from ftp as a
workaround.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ebi.ftp.url</span> <span class="op">&lt;-</span> <span class="st">"ftp.ebi.ac.uk/pub/databases/microarray/data/atlas/experiments"</span>
<span class="va">mtab.url</span> <span class="op">&lt;-</span> <span class="st">"E-MTAB-2770/archive/E-MTAB-2770-atlasExperimentSummary.Rdata.1"</span>
<span class="va">mtab.url</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">ebi.ftp.url</span>, <span class="va">mtab.url</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">mtab.url</span>, <span class="st">"E-MTAB-2770.Rdata"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"E-MTAB-2770.Rdata"</span><span class="op">)</span>
<span class="va">ccle.trans</span> <span class="op">&lt;-</span> <span class="va">experimentSummary</span>
<span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.remove</a></span><span class="op">(</span><span class="st">"E-MTAB-2770.Rdata"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Using cached version from 2022-12-07 01:32:10</span></code></pre>
<p>We proceed as before:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ccle.trans</span> <span class="op">&lt;-</span> <span class="va">ccle.trans</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span>
<span class="va">ccle.trans</span> <span class="op">&lt;-</span> <span class="va">ccle.trans</span><span class="op">[</span>,<span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"HCT 116"</span>, <span class="va">ccle.trans</span><span class="op">$</span><span class="va">cell_line</span><span class="op">)</span><span class="op">]</span>
<span class="va">ccle.trans</span></code></pre></div>
<pre><code><span class="co">## class: RangedSummarizedExperiment </span>
<span class="co">## dim: 58735 1 </span>
<span class="co">## metadata(4): pipeline filtering mapping quantification</span>
<span class="co">## assays(1): counts</span>
<span class="co">## rownames(58735): ENSG00000000003 ENSG00000000005 ... ENSG00000285993</span>
<span class="co">##   ENSG00000285994</span>
<span class="co">## rowData names(0):</span>
<span class="co">## colnames(1): SRR8615282</span>
<span class="co">## colData names(5): AtlasAssayGroup organism cell_line disease</span>
<span class="co">##   organism_part</span></code></pre>
<p>Get the CCLE proteome data for HCT116:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">eh</span> <span class="op">&lt;-</span> <span class="fu">ExperimentHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExperimentHub/man/ExperimentHub-class.html" class="external-link">ExperimentHub</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## snapshotDate(): 2022-10-31</span></code></pre>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">eh</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"gygi"</span>, <span class="st">"depmap"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## ExperimentHub with 1 record</span>
<span class="co">## # snapshotDate(): 2022-10-31</span>
<span class="co">## # names(): EH3459</span>
<span class="co">## # package(): depmap</span>
<span class="co">## # $dataprovider: Broad Institute</span>
<span class="co">## # $species: Homo sapiens</span>
<span class="co">## # $rdataclass: tibble</span>
<span class="co">## # $rdatadateadded: 2020-05-19</span>
<span class="co">## # $title: proteomic_20Q2</span>
<span class="co">## # $description: Quantitative profiling of 12399 proteins in 375 cell lines, ...</span>
<span class="co">## # $taxonomyid: 9606</span>
<span class="co">## # $genome: </span>
<span class="co">## # $sourcetype: CSV</span>
<span class="co">## # $sourceurl: https://gygi.med.harvard.edu/sites/gygi.med.harvard.edu/files/...</span>
<span class="co">## # $sourcesize: NA</span>
<span class="co">## # $tags: c("ExperimentHub", "ExperimentData", "ReproducibleResearch",</span>
<span class="co">## #   "RepositoryData", "AssayDomainData", "CopyNumberVariationData",</span>
<span class="co">## #   "DiseaseModel", "CancerData", "BreastCancerData", "ColonCancerData",</span>
<span class="co">## #   "KidneyCancerData", "LeukemiaCancerData", "LungCancerData",</span>
<span class="co">## #   "OvarianCancerData", "ProstateCancerData", "OrganismData",</span>
<span class="co">## #   "Homo_sapiens_Data", "PackageTypeData", "SpecimenSource",</span>
<span class="co">## #   "CellCulture", "Genome", "Proteome", "StemCell", "Tissue") </span>
<span class="co">## # retrieve record with 'object[["EH3459"]]'</span></code></pre>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ccle.prot</span> <span class="op">&lt;-</span> <span class="va">eh</span><span class="op">[[</span><span class="st">"EH3459"</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<pre><code><span class="co">## 'getOption("repos")' replaces Bioconductor standard repositories, see</span>
<span class="co">## '?repositories' for details</span>
<span class="co">## </span>
<span class="co">## replacement repositories:</span>
<span class="co">##     CRAN: https://packagemanager.rstudio.com/cran/__linux__/focal/2022-06-22</span></code></pre>
<pre><code><span class="co">## Bioconductor version 3.16 (BiocManager 1.30.18), R 4.2.0 (2022-04-22)</span></code></pre>
<pre><code><span class="co">## Installing package(s) 'depmap'</span></code></pre>
<pre><code><span class="co">## Old packages: 'MASS', 'nlme'</span></code></pre>
<pre><code><span class="co">## snapshotDate(): 2022-10-31</span></code></pre>
<pre><code><span class="co">## see ?depmap and browseVignettes('depmap') for documentation</span></code></pre>
<pre><code><span class="co">## loading from cache</span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ccle.prot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">ccle.prot</span><span class="op">)</span>
<span class="va">ccle.prot</span> <span class="op">&lt;-</span> <span class="fu">BioPlex</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioPlex/man/ccleProteome2SummarizedExperiment.html" class="external-link">ccleProteome2SummarizedExperiment</a></span><span class="op">(</span><span class="va">ccle.prot</span><span class="op">)</span>
<span class="va">ccle.prot</span></code></pre></div>
<pre><code><span class="co">## class: SummarizedExperiment </span>
<span class="co">## dim: 12755 1 </span>
<span class="co">## metadata(0):</span>
<span class="co">## assays(1): expr</span>
<span class="co">## rownames(12755): P55011 P35453 ... Q99735 Q9P003</span>
<span class="co">## rowData names(2): SYMBOL ENTREZID</span>
<span class="co">## colnames(1): HCT116</span>
<span class="co">## colData names(0):</span></code></pre>
<p>Connect to <a href="http://bioconductor.org/packages/AnnotationHub" class="external-link">AnnotationHub</a>
and obtain OrgDb package for human:</p>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ah</span> <span class="op">&lt;-</span> <span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">AnnotationHub</a></span><span class="op">(</span><span class="op">)</span>
<span class="va">orgdb</span> <span class="op">&lt;-</span> <span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"orgDb"</span>, <span class="st">"Homo sapiens"</span><span class="op">)</span><span class="op">)</span>
<span class="va">orgdb</span> <span class="op">&lt;-</span> <span class="va">orgdb</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></code></pre></div>
<p>Map to ENSEMBL for comparison with CCLE transcriptome data for
HCT116:</p>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">rnames</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">mapIds</a></span><span class="op">(</span><span class="va">orgdb</span>,
                      keytype <span class="op">=</span> <span class="st">"UNIPROT"</span>,
                      column <span class="op">=</span> <span class="st">"ENSEMBL"</span>,
                      keys <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ccle.prot</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></code></pre>
<p>Subset to the ENSEMBL IDs that both datasets have in common</p>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">isect</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">rnames</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ccle.trans</span><span class="op">)</span><span class="op">)</span>
<span class="va">ind</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">isect</span>, <span class="va">rnames</span><span class="op">)</span></code></pre></div>
<p>This should be rather RPKM, provided gene length from EDASeq:</p>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">ccle.trans</span>, <span class="st">"cpm"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html" class="external-link">cpm</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">ccle.trans</span><span class="op">)</span>, log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<p>A look at general correlation between transcriptome and proteome:</p>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/cor.test.html" class="external-link">cor.test</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">ccle.trans</span>, <span class="st">"cpm"</span><span class="op">)</span><span class="op">[</span><span class="va">isect</span>,<span class="op">]</span>, 
         <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">ccle.prot</span><span class="op">)</span><span class="op">[</span><span class="va">ind</span>,<span class="op">]</span>,
         use <span class="op">=</span> <span class="st">"complete.obs"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##  Pearson's product-moment correlation</span>
<span class="co">## </span>
<span class="co">## data:  assay(ccle.trans, "cpm")[isect, ] and assay(ccle.prot)[ind, ]</span>
<span class="co">## t = 31.482, df = 7843, p-value &lt; 2.2e-16</span>
<span class="co">## alternative hypothesis: true correlation is not equal to 0</span>
<span class="co">## 95 percent confidence interval:</span>
<span class="co">##  0.3151534 0.3544485</span>
<span class="co">## sample estimates:</span>
<span class="co">##       cor </span>
<span class="co">## 0.3349466</span></code></pre>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>trans <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">ccle.trans</span>, <span class="st">"cpm"</span><span class="op">)</span><span class="op">[</span><span class="va">isect</span>,<span class="op">]</span>,
                 prot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">ccle.prot</span><span class="op">)</span><span class="op">[</span><span class="va">ind</span>,<span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">trans</span>, y <span class="op">=</span> <span class="va">prot</span><span class="op">)</span> <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bin_2d.html" class="external-link">geom_bin2d</a></span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">70</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_continuous.html" class="external-link">scale_fill_continuous</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"viridis"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"log2 CPM"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"log2 intensity"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: Removed 2685 rows containing non-finite values (stat_bin2d).</span></code></pre>
<p><img src="TranscriptomeProteome_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>Let’s check whether this looks very different when accounting for
gene length. We therefore obtain gene length for the hg38 genome
assembly (used for CCLE).</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">AnnotationHub</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationHub/man/AnnotationHub-class.html" class="external-link">query</a></span><span class="op">(</span><span class="va">ah</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"TxDb"</span>, <span class="st">"Homo sapiens"</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## AnnotationHub with 39 records</span>
<span class="co">## # snapshotDate(): 2022-10-26</span>
<span class="co">## # $dataprovider: GENCODE, UCSC, NCBI, tRNAdb, snoRNAdb, RMBase v2.0</span>
<span class="co">## # $species: Homo sapiens</span>
<span class="co">## # $rdataclass: TxDb, SQLiteFile, ChainFile, FaFile</span>
<span class="co">## # additional mcols(): taxonomyid, genome, description,</span>
<span class="co">## #   coordinate_1_based, maintainer, rdatadateadded, preparerclass, tags,</span>
<span class="co">## #   rdatapath, sourceurl, sourcetype </span>
<span class="co">## # retrieve records with, e.g., 'object[["AH52256"]]' </span>
<span class="co">## </span>
<span class="co">##              title                                             </span>
<span class="co">##   AH52256  | TxDb.Hsapiens.BioMart.igis.sqlite                 </span>
<span class="co">##   AH52257  | TxDb.Hsapiens.UCSC.hg18.knownGene.sqlite          </span>
<span class="co">##   AH52258  | TxDb.Hsapiens.UCSC.hg19.knownGene.sqlite          </span>
<span class="co">##   AH52259  | TxDb.Hsapiens.UCSC.hg19.lincRNAsTranscripts.sqlite</span>
<span class="co">##   AH52260  | TxDb.Hsapiens.UCSC.hg38.knownGene.sqlite          </span>
<span class="co">##   ...        ...                                               </span>
<span class="co">##   AH92592  | TxDb.Hsapiens.UCSC.hg38.refGene.sqlite            </span>
<span class="co">##   AH97949  | TxDb.Hsapiens.UCSC.hg38.knownGene.sqlite          </span>
<span class="co">##   AH100418 | TxDb.Hsapiens.UCSC.hg38.knownGene.sqlite          </span>
<span class="co">##   AH100419 | TxDb.Hsapiens.UCSC.hg38.refGene.sqlite            </span>
<span class="co">##   AH107068 | TxDb.Hsapiens.UCSC.hg38.knownGene.sqlite</span></code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">txdb</span> <span class="op">&lt;-</span> <span class="va">ah</span><span class="op">[[</span><span class="st">"AH92591"</span><span class="op">]</span><span class="op">]</span>
<span class="va">gs</span> <span class="op">&lt;-</span> <span class="fu">GenomicFeatures</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GenomicFeatures/man/transcripts.html" class="external-link">genes</a></span><span class="op">(</span><span class="va">txdb</span><span class="op">)</span>
<span class="va">gs</span></code></pre></div>
<pre><code><span class="co">## GRanges object with 27113 ranges and 1 metadata column:</span>
<span class="co">##         seqnames              ranges strand |     gene_id</span>
<span class="co">##            &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span>
<span class="co">##       1    chr19   58345178-58362751      - |           1</span>
<span class="co">##      10     chr8   18391282-18401218      + |          10</span>
<span class="co">##     100    chr20   44619522-44652233      - |         100</span>
<span class="co">##    1000    chr18   27932879-28177946      - |        1000</span>
<span class="co">##   10000     chr1 243488233-243851079      - |       10000</span>
<span class="co">##     ...      ...                 ...    ... .         ...</span>
<span class="co">##    9991     chr9 112217716-112333664      - |        9991</span>
<span class="co">##    9992    chr21   34364006-34371381      + |        9992</span>
<span class="co">##    9993    chr22   19036282-19122454      - |        9993</span>
<span class="co">##    9994     chr6   89829894-89874436      + |        9994</span>
<span class="co">##    9997    chr22   50523568-50526461      - |        9997</span>
<span class="co">##   -------</span>
<span class="co">##   seqinfo: 595 sequences (1 circular) from hg38 genome</span></code></pre>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">len</span> <span class="op">&lt;-</span> <span class="fu">GenomicRanges</span><span class="fu">::</span><span class="fu">width</span><span class="op">(</span><span class="va">gs</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">len</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">gs</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">len</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##         1        10       100      1000     10000 100009613 </span>
<span class="co">##     17574      9937     32712    245068    362847      3000</span></code></pre>
<p>This requires to map from Entrez IDs present for the gene length data
to ENSEMBL IDs present in the transcriptomic data.</p>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">eids</span> <span class="op">&lt;-</span> <span class="fu">AnnotationDbi</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">mapIds</a></span><span class="op">(</span><span class="va">orgdb</span>,
                              column <span class="op">=</span> <span class="st">"ENTREZID"</span>,
                              keytype <span class="op">=</span> <span class="st">"ENSEMBL"</span>, 
                              keys <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ccle.trans</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">ccle.trans</span><span class="op">)</span><span class="op">$</span><span class="va">length</span> <span class="op">&lt;-</span> <span class="va">len</span><span class="op">[</span><span class="va">eids</span><span class="op">]</span></code></pre></div>
<p>We can now compute RPKM given the obtained gene lengths as input.</p>
<div class="sourceCode" id="cb49"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">ccle.trans</span>, <span class="st">"rpkm"</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html" class="external-link">rpkm</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">ccle.trans</span><span class="op">)</span>, 
                                         gene.length <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">ccle.trans</span><span class="op">)</span><span class="op">$</span><span class="va">length</span>,
                                         log <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> </code></pre></div>
<div class="sourceCode" id="cb50"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/cor.test.html" class="external-link">cor.test</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">ccle.trans</span>, <span class="st">"rpkm"</span><span class="op">)</span><span class="op">[</span><span class="va">isect</span>,<span class="op">]</span>,
         <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">ccle.prot</span><span class="op">)</span><span class="op">[</span><span class="va">ind</span>,<span class="op">]</span>,
         use <span class="op">=</span> <span class="st">"complete.obs"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##  Pearson's product-moment correlation</span>
<span class="co">## </span>
<span class="co">## data:  assay(ccle.trans, "rpkm")[isect, ] and assay(ccle.prot)[ind, ]</span>
<span class="co">## t = 29.516, df = 7811, p-value &lt; 2.2e-16</span>
<span class="co">## alternative hypothesis: true correlation is not equal to 0</span>
<span class="co">## 95 percent confidence interval:</span>
<span class="co">##  0.2966832 0.3365837</span>
<span class="co">## sample estimates:</span>
<span class="co">##       cor </span>
<span class="co">## 0.3167736</span></code></pre>
<div class="sourceCode" id="cb52"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>trans <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">ccle.trans</span>, <span class="st">"rpkm"</span><span class="op">)</span><span class="op">[</span><span class="va">isect</span>,<span class="op">]</span>,
                 prot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">ccle.prot</span><span class="op">)</span><span class="op">[</span><span class="va">ind</span>,<span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">trans</span>, y <span class="op">=</span> <span class="va">prot</span><span class="op">)</span> <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bin_2d.html" class="external-link">geom_bin2d</a></span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">70</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_continuous.html" class="external-link">scale_fill_continuous</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"viridis"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"log2 RPKM"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"log2 intensity"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Warning: Removed 2717 rows containing non-finite values (stat_bin2d).</span></code></pre>
<p><img src="TranscriptomeProteome_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="de-analysis-hek293-vs--hct116-transcriptomic-and-proteomic-level">DE analysis HEK293 vs. HCT116 (transcriptomic and proteomic
level)<a class="anchor" aria-label="anchor" href="#de-analysis-hek293-vs--hct116-transcriptomic-and-proteomic-level"></a>
</h2>
<p>Pull the HEK293 data:</p>
<div class="sourceCode" id="cb54"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gse.293t</span> <span class="op">&lt;-</span> <span class="fu">BioPlex</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioPlex/man/getGSE122425.html" class="external-link">getGSE122425</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Using cached version from 2022-12-07 01:06:20</span></code></pre>
<p>Pull the HCT116 data:</p>
<div class="sourceCode" id="cb56"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">klijn</span> <span class="op">&lt;-</span> <span class="fu">ExpressionAtlas</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ExpressionAtlas/man/getAtlasData.html" class="external-link">getAtlasData</a></span><span class="op">(</span><span class="st">"E-MTAB-2706"</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Downloading Expression Atlas experiment summary from:</span>
<span class="co">##  ftp://ftp.ebi.ac.uk/pub/databases/microarray/data/atlas/experiments/E-MTAB-2706/E-MTAB-2706-atlasExperimentSummary.Rdata</span></code></pre>
<pre><code><span class="co">## Successfully downloaded experiment summary object for E-MTAB-2706</span></code></pre>
<div class="sourceCode" id="cb59"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">klijn</span> <span class="op">&lt;-</span> <span class="va">klijn</span><span class="op">$</span><span class="va">`E-MTAB-2706`</span><span class="op">$</span><span class="va">rnaseq</span>
<span class="va">klijn</span></code></pre></div>
<pre><code><span class="co">## class: RangedSummarizedExperiment </span>
<span class="co">## dim: 65217 622 </span>
<span class="co">## metadata(4): pipeline filtering mapping quantification</span>
<span class="co">## assays(1): counts</span>
<span class="co">## rownames(65217): ENSG00000000003 ENSG00000000005 ... ENSG00000281921</span>
<span class="co">##   ENSG00000281922</span>
<span class="co">## rowData names(0):</span>
<span class="co">## colnames(622): ERR413347 ERR413348 ... ERR414020 ERR415514</span>
<span class="co">## colData names(12): AtlasAssayGroup organism ... media freeze_media</span></code></pre>
<p>Combine the both HCT116 samples:</p>
<div class="sourceCode" id="cb61"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ind2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"HCT 116"</span>, <span class="va">klijn</span><span class="op">$</span><span class="va">cell_line</span><span class="op">)</span>
<span class="va">isect</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">ccle.trans</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">klijn</span><span class="op">)</span><span class="op">)</span>
<span class="va">emat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">ccle.trans</span><span class="op">)</span><span class="op">[</span><span class="va">isect</span>,<span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">klijn</span><span class="op">)</span><span class="op">[</span><span class="va">isect</span>,<span class="va">ind2</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">emat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ccle"</span>, <span class="st">"klijn"</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">emat</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##                 ccle klijn</span>
<span class="co">## ENSG00000000003 2441  1876</span>
<span class="co">## ENSG00000000005    0     0</span>
<span class="co">## ENSG00000000419 3950  3731</span>
<span class="co">## ENSG00000000457 1085   676</span>
<span class="co">## ENSG00000000460 1680  1206</span>
<span class="co">## ENSG00000000938    0     0</span></code></pre>
<p>Combine with the HEK293 wildtype samples:</p>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">isect</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">emat</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gse.293t</span><span class="op">)</span><span class="op">)</span>
<span class="va">emat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">emat</span><span class="op">[</span><span class="va">isect</span>,<span class="op">]</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">gse.293t</span><span class="op">)</span><span class="op">[</span><span class="va">isect</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">emat</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HCT"</span>, <span class="st">"HEK"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">3</span><span class="op">)</span><span class="op">)</span> </code></pre></div>
<p>Compute logCPMs to bring samples from different cell lines and
experiments on the same scale using the limma-trend approach:</p>
<div class="sourceCode" id="cb64"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dge</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/DGEList.html" class="external-link">DGEList</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="va">emat</span><span class="op">)</span>
<span class="va">dge</span><span class="op">$</span><span class="va">group</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">rep</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"HCT"</span>, <span class="st">"HEK"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/pkg/S4Vectors/man/Vector-class.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span><span class="op">)</span><span class="op">)</span>
<span class="va">design</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/model.matrix.html" class="external-link">model.matrix</a></span><span class="op">(</span><span class="op">~</span> <span class="va">dge</span><span class="op">$</span><span class="va">group</span><span class="op">)</span>
<span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/filterByExpr.html" class="external-link">filterByExpr</a></span><span class="op">(</span><span class="va">dge</span>, <span class="va">design</span><span class="op">)</span>
<span class="va">dge</span> <span class="op">&lt;-</span> <span class="va">dge</span><span class="op">[</span><span class="va">keep</span>,,keep.lib.sizes <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span>
<span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">dge</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 19186     5</span></code></pre>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">dge</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/calcNormFactors.html" class="external-link">calcNormFactors</a></span><span class="op">(</span><span class="va">dge</span><span class="op">)</span>
<span class="va">logCPM</span> <span class="op">&lt;-</span> <span class="fu">edgeR</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/edgeR/man/cpm.html" class="external-link">cpm</a></span><span class="op">(</span><span class="va">dge</span>, log <span class="op">=</span> <span class="cn">TRUE</span>, prior.count <span class="op">=</span> <span class="fl">3</span><span class="op">)</span>
<span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/lmFit.html" class="external-link">lmFit</a></span><span class="op">(</span><span class="va">logCPM</span>, <span class="va">design</span><span class="op">)</span>
<span class="va">fit</span> <span class="op">&lt;-</span> <span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/ebayes.html" class="external-link">eBayes</a></span><span class="op">(</span><span class="va">fit</span>, trend <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>
<span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">##                      logFC  AveExpr         t      P.Value    adj.P.Val</span>
<span class="co">## ENSG00000176788   9.927080 4.200182  92.21793 5.841381e-10 3.567027e-06</span>
<span class="co">## ENSG00000134871   9.242679 4.650104  88.41680 7.356481e-10 3.567027e-06</span>
<span class="co">## ENSG00000198786 -15.839842 2.433343 -83.70579 9.929857e-10 3.567027e-06</span>
<span class="co">## ENSG00000261409  12.811237 3.993704  80.88389 1.198204e-09 3.567027e-06</span>
<span class="co">## ENSG00000133124  12.765158 3.940417  80.19682 1.255526e-09 3.567027e-06</span>
<span class="co">## ENSG00000198695 -13.775054 1.607428 -79.93320 1.278376e-09 3.567027e-06</span>
<span class="co">## ENSG00000159217   8.156433 4.811616  79.67285 1.301428e-09 3.567027e-06</span>
<span class="co">## ENSG00000041982   8.146714 4.445960  74.71939 1.849773e-09 3.963810e-06</span>
<span class="co">## ENSG00000181291  10.457757 2.372061  74.64867 1.859392e-09 3.963810e-06</span>
<span class="co">## ENSG00000138829   9.380895 4.616683  70.70227 2.503608e-09 4.342763e-06</span>
<span class="co">##                        B</span>
<span class="co">## ENSG00000176788 11.63254</span>
<span class="co">## ENSG00000134871 11.56147</span>
<span class="co">## ENSG00000198786 11.46225</span>
<span class="co">## ENSG00000261409 11.39606</span>
<span class="co">## ENSG00000133124 11.37910</span>
<span class="co">## ENSG00000198695 11.37249</span>
<span class="co">## ENSG00000159217 11.36592</span>
<span class="co">## ENSG00000041982 11.23057</span>
<span class="co">## ENSG00000181291 11.22849</span>
<span class="co">## ENSG00000138829 11.10450</span></code></pre>
<div class="sourceCode" id="cb68"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">tt</span> <span class="op">&lt;-</span> <span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/toptable.html" class="external-link">topTable</a></span><span class="op">(</span><span class="va">fit</span>, coef <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">design</span><span class="op">)</span>, number <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-toTable.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">logCPM</span><span class="op">)</span>, sort.by <span class="op">=</span> <span class="st">"none"</span><span class="op">)</span></code></pre></div>
<p>Now let’s pull the BioPlex3 proteome data:</p>
<div class="sourceCode" id="cb69"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">bp.prot</span> <span class="op">&lt;-</span> <span class="fu">BioPlex</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BioPlex/man/getBioplexProteome.html" class="external-link">getBioplexProteome</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## Using cached version from 2022-12-07 01:26:48</span></code></pre>
<div class="sourceCode" id="cb71"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## DataFrame with 9604 rows and 5 columns</span>
<span class="co">##             ENTREZID      SYMBOL nr.peptides log2ratio  adj.pvalue</span>
<span class="co">##          &lt;character&gt; &lt;character&gt;   &lt;integer&gt; &lt;numeric&gt;   &lt;numeric&gt;</span>
<span class="co">## P0CG40     100131390         SP9           1 -2.819071 6.66209e-08</span>
<span class="co">## Q8IXZ3-4      221833         SP8           3 -3.419888 6.94973e-07</span>
<span class="co">## P55011          6558     SLC12A2           4  0.612380 4.85602e-06</span>
<span class="co">## O60341         23028       KDM1A           7 -0.319695 5.08667e-04</span>
<span class="co">## O14654          8471        IRS4           4 -5.951096 1.45902e-06</span>
<span class="co">## ...              ...         ...         ...       ...         ...</span>
<span class="co">## Q9H6X4         80194     TMEM134           2 -0.379342 7.67195e-05</span>
<span class="co">## Q9BS91         55032     SLC35A5           1 -2.237634 8.75523e-05</span>
<span class="co">## Q9UKJ5         26511       CHIC2           1 -0.614932 1.78756e-03</span>
<span class="co">## Q9H3S5         93183        PIGM           1 -1.011397 8.91589e-06</span>
<span class="co">## Q8WYQ3        400916     CHCHD10           1  0.743852 1.17163e-03</span></code></pre>
<p>Compare differential expression results on transcriptomic and
proteomic level based on gene symbols as those are readily
available:</p>
<div class="sourceCode" id="cb73"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">isect</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/sets.html" class="external-link">intersect</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span><span class="op">$</span><span class="va">SYMBOL</span>, 
                   <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">gse.293t</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">logCPM</span><span class="op">)</span>, <span class="st">"SYMBOL"</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">isect</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## [1] 8974</span></code></pre>
<div class="sourceCode" id="cb75"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">ind.trans</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">isect</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">gse.293t</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">logCPM</span><span class="op">)</span>, <span class="st">"SYMBOL"</span><span class="op">]</span><span class="op">)</span>
<span class="va">ind.prot</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">isect</span>, <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span><span class="op">$</span><span class="va">SYMBOL</span><span class="op">)</span></code></pre></div>
<p>We need to switch here the sign of the fold change because the
transcriptome is HEK-vs-HCT, the proteome is HCT-vs-HEK:</p>
<div class="sourceCode" id="cb76"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/stats/cor.test.html" class="external-link">cor.test</a></span><span class="op">(</span><span class="op">-</span><span class="fl">1</span> <span class="op">*</span> <span class="va">tt</span><span class="op">[</span><span class="va">ind.trans</span>, <span class="st">"logFC"</span><span class="op">]</span>, 
         <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span><span class="op">[</span><span class="va">ind.prot</span>, <span class="st">"log2ratio"</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<pre><code><span class="co">## </span>
<span class="co">##  Pearson's product-moment correlation</span>
<span class="co">## </span>
<span class="co">## data:  -1 * tt[ind.trans, "logFC"] and rowData(bp.prot)[ind.prot, "log2ratio"]</span>
<span class="co">## t = 70.463, df = 8972, p-value &lt; 2.2e-16</span>
<span class="co">## alternative hypothesis: true correlation is not equal to 0</span>
<span class="co">## 95 percent confidence interval:</span>
<span class="co">##  0.5833789 0.6100218</span>
<span class="co">## sample estimates:</span>
<span class="co">##       cor </span>
<span class="co">## 0.5968649</span></code></pre>
<div class="sourceCode" id="cb78"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>trans <span class="op">=</span> <span class="op">-</span><span class="fl">1</span> <span class="op">*</span> <span class="va">tt</span><span class="op">[</span><span class="va">ind.trans</span>, <span class="st">"logFC"</span><span class="op">]</span>,
                 prot <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">rowData</a></span><span class="op">(</span><span class="va">bp.prot</span><span class="op">)</span><span class="op">[</span><span class="va">ind.prot</span>, <span class="st">"log2ratio"</span><span class="op">]</span><span class="op">)</span>
<span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">trans</span>, y <span class="op">=</span> <span class="va">prot</span><span class="op">)</span> <span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bin_2d.html" class="external-link">geom_bin2d</a></span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">70</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_colour_continuous.html" class="external-link">scale_fill_continuous</a></span><span class="op">(</span>type <span class="op">=</span> <span class="st">"viridis"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">xlab</a></span><span class="op">(</span><span class="st">"log2FC (transcriptome)"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"log2FC (proteome)"</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span></code></pre></div>
<p><img src="TranscriptomeProteome_files/figure-html/unnamed-chunk-24-1.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Ludwig Geistlinger, Robert Gentleman.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.4.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
